diff -p -up gdm-2.20.6/daemon/gdm-daemon-config.c.fixfree gdm-2.20.6/daemon/gdm-daemon-config.c
--- gdm-2.20.6/daemon/gdm-daemon-config.c.fixfree	2008-06-03 11:41:15.000000000 +0200
+++ gdm-2.20.6/daemon/gdm-daemon-config.c	2008-06-03 11:41:17.000000000 +0200
@@ -1433,7 +1433,7 @@ gdm_daemon_config_load_displays (GdmConf
 		g_strfreev (value_list);
 	}
 
-	g_ptr_array_free ((GPtrArray*) keys, TRUE);
+	g_free (keys);
 }
 
 static gboolean
