GNOME #649413

Index: gdm-3.0.0/daemon/gdm-session.c
===================================================================
--- gdm-3.0.0.orig/daemon/gdm-session.c	2011-04-29 01:19:33.240497348 +0200
+++ gdm-3.0.0/daemon/gdm-session.c	2011-04-29 01:19:39.568528281 +0200
@@ -1970,9 +1970,6 @@ setup_session_environment (GdmSessionDir
         locale = get_language_name (self);
 
         if (locale != NULL && locale[0] != '\0') {
-                gdm_session_set_environment_variable (self,
-                                                      "LANG",
-                                                      locale);
                 gdm_session_set_environment_variable (self,
                                                       "GDM_LANG",
                                                       locale);

