Index: gdm-2.26.1/configure.ac
===================================================================
--- gdm-2.26.1.orig/configure.ac	2009-03-16 20:50:10.000000000 +0100
+++ gdm-2.26.1/configure.ac	2009-07-14 18:25:15.021281016 +0200
@@ -1146,28 +1146,10 @@ dnl ------------------------------------
 dnl - Check for Xnest / Xephyr support
 dnl ---------------------------------------------------------------------------
 
-# Use Xephyr if it is available.  It works better than Xnest since Xephyr
-# supports the Xserver extensions, even if on a remote machine.
-#
+# Use the wrapper script for the nested X server
+X_XNEST_CMD=/usr/share/gdm/gdmXnestWrapper
+X_XNEST_CONFIG_OPTIONS="-audit 0"
 X_XNEST_UNSCALED_FONTPATH="true"
-if test -x $X_SERVER_PATH/Xephyr; then
-    X_XNEST_CMD="$X_SERVER_PATH/Xephyr"
-    X_XNEST_CONFIG_OPTIONS="-audit 0"
-else
-    if test -x /usr/openwin/bin/Xnest; then
-        # If on Solaris, Xnest is only shipped with the Xsun Xserver, so
-        # use this version.
-        #
-        X_XNEST_CMD="/usr/openwin/bin/Xnest"
-        X_XNEST_CONFIG_OPTIONS="-audit 0 -name Xnest -pn"
-        X_XNEST_UNSCALED_FONTPATH="false"
-    else
-        if test -x $X_SERVER_PATH/Xnest; then
-            X_XNEST_CMD="$X_SERVER_PATH/Xnest"
-            X_XNEST_CONFIG_OPTIONS="-audit 0 -name Xnest"
-        fi
-    fi
-fi
 
 dnl ---------------------------------------------------------------------------
 dnl - Expand vars
