--- a/configure.ac
+++ b/configure.ac
@@ -946,7 +946,7 @@
 dnl - Define some variables to represent the directories we use.
 dnl ---------------------------------------------------------------------------
 
-AC_SUBST(authdir, ${localstatedir}/gdm)
+AC_SUBST(authdir, ${localstatedir}/lib/gdm)
 AC_SUBST(gdmlocaledir, ${gdmconfdir})
 AC_SUBST(pixmapdir, ${datadir}/pixmaps)
 
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -7,7 +7,7 @@
 	$(NULL)
 
 initdir = $(gdmconfdir)/Init
-authdir = $(localstatedir)/gdm
+authdir = $(localstatedir)/lib/gdm
 postdir = $(gdmconfdir)/PostSession
 predir = $(gdmconfdir)/PreSession
 postlogindir = $(gdmconfdir)/PostLogin
