diff -urN gdm-2.28.1.orig/common/gdm-settings-desktop-backend.c gdm-2.28.1/common/gdm-settings-desktop-backend.c
--- gdm-2.28.1.orig/common/gdm-settings-desktop-backend.c	2009-10-20 00:12:45.000000000 +0200
+++ gdm-2.28.1/common/gdm-settings-desktop-backend.c	2009-10-25 11:30:12.000000000 +0100
@@ -327,7 +327,7 @@
         backend->priv = GDM_SETTINGS_DESKTOP_BACKEND_GET_PRIVATE (backend);
 
         backend->priv->key_file = g_key_file_new ();
-        backend->priv->filename = g_strdup (GDMCONFDIR "/custom.conf");
+        backend->priv->filename = g_strdup (GDM_CUSTOM_CONF);
 
         error = NULL;
         res = g_key_file_load_from_file (backend->priv->key_file,
