Index: gdm-2.30.0/daemon/gdm-welcome-session.c
===================================================================
--- gdm-2.30.0.orig/daemon/gdm-welcome-session.c	2010-04-13 23:55:36.713237666 +0200
+++ gdm-2.30.0/daemon/gdm-welcome-session.c	2010-04-14 00:05:56.241737918 +0200
@@ -46,7 +46,7 @@
 
 #include "gdm-welcome-session.h"
 
-#define DBUS_LAUNCH_COMMAND BINDIR "/dbus-launch --exit-with-session"
+#define DBUS_LAUNCH_COMMAND BINDIR "/dbus-launch"
 
 #define MAX_LOGS 5
 
