Description: Run /etc/X11/Xsession.d/.
Author: Martin Pitt <martin.pitt@ubuntu.com>
Origin: Ubuntu; /etc/X11/Xsession.d is a Debianism/Ubuntuism

--- gdm-2.30.2/data/Xsession.in	2010-03-26 14:28:03.000000000 +0100
+++ gdm-2.30.2.new/data/Xsession.in	2010-04-28 13:17:28.266395400 +0200
@@ -186,6 +186,19 @@
     done
 fi
 
+# run /etc/X11/Xsession.d/
+OPTIONFILE=/etc/X11/Xsession.options
+USERXSESSION=$HOME/.xsession
+USERXSESSIONRC=$HOME/.xsessionrc
+ALTUSERXSESSION=$HOME/.Xsession
+if [ -d /etc/X11/Xsession.d ]; then
+    for i in `ls /etc/X11/Xsession.d/` ; do
+        if [ -r "/etc/X11/Xsession.d/$i" ] && expr "$i" : '^[[:alnum:]_-]\+$' > /dev/null; then
+	    . "/etc/X11/Xsession.d/$i"
+        fi
+    done
+fi
+
 if [ "x$command" = "xcustom" ] ; then
   if [ -x "$HOME/.xsession" ]; then
     command="$HOME/.xsession"
