Description: use lsb informations to correctly display the distribution
Bug: http://bugzilla.gnome.org/show_bug.cgi?id=589549
Bug-Ubuntu: https://bugs.launchpad.net/bugs/396805

Index: gdm-2.30.2/gui/simple-greeter/gdm-greeter-login-window.c
===================================================================
--- gdm-2.30.2.orig/gui/simple-greeter/gdm-greeter-login-window.c	2010-04-27 03:08:42.000000000 +1000
+++ gdm-2.30.2/gui/simple-greeter/gdm-greeter-login-window.c	2010-06-22 13:10:49.934144597 +1000
@@ -1022,10 +1022,16 @@
 static char *
 get_system_version (void)
 {
-        char *version;
+        char *version, *output;
         int i;
 
         version = NULL;
+        output = NULL;                
+
+        if (g_spawn_command_line_sync ("lsb_release -sd", &output, NULL, NULL, NULL)) {
+                version = g_strchomp (output);
+                return version;
+        }    
 
         for (i = 0; known_etc_info_files [i]; i++) {
                 char *path1;
@@ -1046,8 +1052,6 @@
         }
 
         if (version == NULL) {
-                char *output;
-                output = NULL;
                 if (g_spawn_command_line_sync ("uname -sr", &output, NULL, NULL, NULL)) {
                         version = g_strchomp (output);
                 }
