GNOME #645918

Index: gdm-3.0.4/daemon/gdm-server.c
===================================================================
--- gdm-3.0.4.orig/daemon/gdm-server.c	2011-05-31 23:59:46.000000000 +0200
+++ gdm-3.0.4/daemon/gdm-server.c	2011-06-01 00:00:15.501219261 +0200
@@ -134,7 +134,7 @@ _gdm_server_query_ck_for_display_device
         g_return_val_if_fail (GDM_IS_SERVER (server), NULL);
 
         error = NULL;
-        command = g_strdup_printf (LIBEXECDIR "/ck-get-x11-display-device --display %s",
+        command = g_strdup_printf (CK_LIBEXECDIR "/ck-get-x11-display-device --display %s",
                                    server->priv->display_name);
 
         g_debug ("GdmServer: Running helper %s", command);
Index: gdm-3.0.4/data/greeter-autostart/gnome-settings-daemon.desktop.in.in
===================================================================
--- gdm-3.0.4.orig/data/greeter-autostart/gnome-settings-daemon.desktop.in.in	2011-05-31 17:46:56.000000000 +0200
+++ gdm-3.0.4/data/greeter-autostart/gnome-settings-daemon.desktop.in.in	2011-06-01 00:00:15.501219261 +0200
@@ -1,8 +1,8 @@
 [Desktop Entry]
 Type=Application
 _Name=GNOME Settings Daemon
-TryExec=@LIBEXECDIR@/gnome-settings-daemon
-Exec=@LIBEXECDIR@/gnome-settings-daemon
+TryExec=@GSD_LIBEXECDIR@/gnome-settings-daemon
+Exec=@GSD_LIBEXECDIR@/gnome-settings-daemon
 OnlyShowIn=GNOME;
 X-GNOME-Autostart-Phase=Initialization
 X-GNOME-Autostart-Notify=true
Index: gdm-3.0.4/gui/simple-chooser/gdm-chooser-session.c
===================================================================
--- gdm-3.0.4.orig/gui/simple-chooser/gdm-chooser-session.c	2011-05-31 17:46:56.000000000 +0200
+++ gdm-3.0.4/gui/simple-chooser/gdm-chooser-session.c	2011-06-01 00:00:15.501219261 +0200
@@ -131,7 +131,7 @@ start_settings_daemon (GdmChooserSession
         ret = FALSE;
 
         error = NULL;
-        g_spawn_command_line_async (LIBEXECDIR "/gnome-settings-daemon", &error);
+        g_spawn_command_line_async (GSD_LIBEXECDIR "/gnome-settings-daemon", &error);
         if (error != NULL) {
                 g_warning ("Error starting settings daemon: %s", error->message);
                 g_error_free (error);
Index: gdm-3.0.4/data/greeter-autostart/polkit-gnome-authentication-agent-1.desktop.in.in
===================================================================
--- gdm-3.0.4.orig/data/greeter-autostart/polkit-gnome-authentication-agent-1.desktop.in.in	2011-03-07 18:40:37.000000000 +0100
+++ gdm-3.0.4/data/greeter-autostart/polkit-gnome-authentication-agent-1.desktop.in.in	2011-06-01 00:00:15.501219261 +0200
@@ -1,5 +1,5 @@
 [Desktop Entry]
 Type=Application
 _Name=PolicyKit Authentication Agent
-Exec=@LIBEXECDIR@/polkit-gnome-authentication-agent-1
+Exec=@PK_GNOME_LIBEXECDIR@/polkit-gnome-authentication-agent-1
 OnlyShowIn=GNOME;
Index: gdm-3.0.4/configure.ac
===================================================================
--- gdm-3.0.4.orig/configure.ac	2011-05-31 23:59:43.000000000 +0200
+++ gdm-3.0.4/configure.ac	2011-06-01 00:00:32.821303957 +0200
@@ -52,6 +52,7 @@ LIBCANBERRA_GTK_REQUIRED_VERSION=0.4
 FONTCONFIG_REQUIRED_VERSION=2.5.0
 UPOWER_REQUIRED_VERSION=0.9.0
 ACCOUNTS_SERVICE_REQUIRED_VERSION=0.6.12
+GSD_REQUIRED_VERSION=2.91.2
 
 EXTRA_COMPILE_WARNINGS(yes)
 
@@ -170,6 +171,39 @@ AC_PATH_PROG(GCONFTOOL, gconftool-2)
 AM_GCONF_SOURCE_2
 
 dnl ---------------------------------------------------------------------------
+dnl - Paths for dependencies
+dnl ---------------------------------------------------------------------------
+
+PKG_CHECK_MODULES(GSD,
+        gnome-settings-daemon >= $GSD_REQUIRED_VERSION
+)
+GSD_LIBEXECDIR=`pkg-config --variable=libexecdir gnome-settings-daemon`
+AC_SUBST(GSD_LIBEXECDIR)
+AC_DEFINE_UNQUOTED([GSD_LIBEXECDIR], "$GSD_LIBEXECDIR", [Path to the GNOME settings daemon])
+
+
+dnl This should be done with pkg-config, but ConsoleKit and PolicyKit don’t
+dnl ship .pc files
+dnl Because of that we only handle the case where libexecdir is the same
+dnl (e.g. Red Hat), and the case where libexecdir = pkglibdir (e.g. Debian)
+
+AC_MSG_CHECKING(where helpers are installed)
+if test x`basename "$libexecdir"` = "xlibexec"; then
+        AC_MSG_RESULT($libexecdir)
+        CK_LIBEXECDIR="$libexecdir"
+        PK_GNOME_LIBEXECDIR="$libexecdir"
+        SESSION_LIBEXECDIR="$libexecdir"
+else
+        AC_MSG_RESULT("$libdir/Consolekit, $libdir/policykit-1-gnome, $libdir/gnome-session")
+        CK_LIBEXECDIR="$libdir/ConsoleKit"
+        PK_GNOME_LIBEXECDIR="$libdir/policykit-1-gnome"
+        SESSION_LIBEXECDIR="$libdir/gnome-session"
+fi
+AC_SUBST(CK_LIBEXECDIR)
+AC_SUBST(PK_GNOME_LIBEXECDIR)
+AC_SUBST(SESSION_LIBEXECDIR)
+
+dnl ---------------------------------------------------------------------------
 dnl - Configuration file stuff
 dnl ---------------------------------------------------------------------------
 
Index: gdm-3.0.4/daemon/Makefile.am
===================================================================
--- gdm-3.0.4.orig/daemon/Makefile.am	2011-05-31 16:28:03.000000000 +0200
+++ gdm-3.0.4/daemon/Makefile.am	2011-06-01 00:00:15.501219261 +0200
@@ -12,6 +12,7 @@ AM_CPPFLAGS = \
 	-DGDMCONFDIR=\"$(gdmconfdir)\"			\
 	-DLIBDIR=\"$(libdir)\"				\
 	-DLIBEXECDIR=\"$(libexecdir)\"			\
+	-DCK_LIBEXECDIR=\"$(CK_LIBEXECDIR)\"			\
 	-DLOGDIR=\"$(logdir)\"				\
 	-DSBINDIR=\"$(sbindir)\"			\
 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"	\
Index: gdm-3.0.4/data/greeter-autostart/Makefile.am
===================================================================
--- gdm-3.0.4.orig/data/greeter-autostart/Makefile.am	2011-05-31 17:46:56.000000000 +0200
+++ gdm-3.0.4/data/greeter-autostart/Makefile.am	2011-06-01 00:00:15.501219261 +0200
@@ -5,6 +5,9 @@ edit = sed \
 	-e 's|@SYSCONFDIR[@]|$(sysconfdir)|g' \
 	-e 's|@LIBDIR[@]|$(libdir)|g' \
 	-e 's|@LIBEXECDIR[@]|$(libexecdir)|g' \
+	-e 's|@GSD_LIBEXECDIR[@]|$(GSD_LIBEXECDIR)|g' \
+	-e 's|@PK_GNOME_LIBEXECDIR[@]|$(PK_GNOME_LIBEXECDIR)|g' \
+	-e 's|@SESSION_LIBEXECDIR[@]|$(SESSION_LIBEXECDIR)|g' \
 	-e 's|@LOCALSTATEDIR[@]|$(localstatedir)|g'
 
 %.desktop.in: %.desktop.in.in Makefile
Index: gdm-3.0.4/data/greeter-autostart/gnome-session-check-accelerated.desktop.in.in
===================================================================
--- gdm-3.0.4.orig/data/greeter-autostart/gnome-session-check-accelerated.desktop.in.in	2011-05-31 17:46:56.000000000 +0200
+++ gdm-3.0.4/data/greeter-autostart/gnome-session-check-accelerated.desktop.in.in	2011-06-01 00:00:15.501219261 +0200
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Type=Application
 _Name=GNOME Session Acceleration Checker
-TryExec=@LIBEXECDIR@/gnome-session-check-accelerated
-Exec=@LIBEXECDIR@/gnome-session-check-accelerated
+TryExec=@SESSION_LIBEXECDIR@/gnome-session-check-accelerated
+Exec=@SESSION_LIBEXECDIR@/gnome-session-check-accelerated
 X-GNOME-Autostart-Phase=Application
