GNOME #645918

Index: gdm-3.2.1.1/daemon/gdm-server.c
===================================================================
--- gdm-3.2.1.1.orig/daemon/gdm-server.c	2011-12-17 03:25:47.000000000 +0100
+++ gdm-3.2.1.1/daemon/gdm-server.c	2011-12-17 03:27:51.248986991 +0100
@@ -134,7 +134,7 @@ _gdm_server_query_ck_for_display_device
         g_return_val_if_fail (GDM_IS_SERVER (server), NULL);
 
         error = NULL;
-        command = g_strdup_printf (LIBEXECDIR "/ck-get-x11-display-device --display %s",
+        command = g_strdup_printf (CK_LIBEXECDIR "/ck-get-x11-display-device --display %s",
                                    server->priv->display_name);
 
         g_debug ("GdmServer: Running helper %s", command);
Index: gdm-3.2.1.1/gui/simple-chooser/gdm-chooser-session.c
===================================================================
--- gdm-3.2.1.1.orig/gui/simple-chooser/gdm-chooser-session.c	2011-10-19 17:41:52.000000000 +0200
+++ gdm-3.2.1.1/gui/simple-chooser/gdm-chooser-session.c	2011-12-17 03:27:51.248986991 +0100
@@ -131,7 +131,7 @@ start_settings_daemon (GdmChooserSession
         ret = FALSE;
 
         error = NULL;
-        g_spawn_command_line_async (LIBEXECDIR "/gnome-settings-daemon", &error);
+        g_spawn_command_line_async (GSD_LIBEXECDIR "/gnome-settings-daemon", &error);
         if (error != NULL) {
                 g_warning ("Error starting settings daemon: %s", error->message);
                 g_error_free (error);
Index: gdm-3.2.1.1/configure.ac
===================================================================
--- gdm-3.2.1.1.orig/configure.ac	2011-12-17 03:24:25.000000000 +0100
+++ gdm-3.2.1.1/configure.ac	2011-12-17 03:49:29.831335428 +0100
@@ -68,6 +68,7 @@ LIBCANBERRA_GTK_REQUIRED_VERSION=0.4
 FONTCONFIG_REQUIRED_VERSION=2.5.0
 UPOWER_REQUIRED_VERSION=0.9.0
 ACCOUNTS_SERVICE_REQUIRED_VERSION=0.6.12
+GSD_REQUIRED_VERSION=2.91.2
 NSS_REQUIRED_VERSION=3.11.1
 
 EXTRA_COMPILE_WARNINGS(yes)
@@ -203,6 +204,33 @@ LIBGDMGREETER_GIR_INCLUDES="GLib-2.0 GOb
 AC_SUBST(LIBGDMGREETER_GIR_INCLUDES)
 
 dnl ---------------------------------------------------------------------------
+dnl - Paths for dependencies
+dnl ---------------------------------------------------------------------------
+
+PKG_CHECK_MODULES(GSD,
+        gnome-settings-daemon >= $GSD_REQUIRED_VERSION
+)
+GSD_LIBEXECDIR=`pkg-config --variable=libexecdir gnome-settings-daemon`
+AC_SUBST(GSD_LIBEXECDIR)
+AC_DEFINE_UNQUOTED([GSD_LIBEXECDIR], "$GSD_LIBEXECDIR", [Path to the GNOME settings daemon])
+
+
+dnl This should be done with pkg-config, but ConsoleKit and PolicyKit don’t
+dnl ship .pc files
+dnl Because of that we only handle the case where libexecdir is the same
+dnl (e.g. Red Hat), and the case where libexecdir = pkglibdir (e.g. Debian)
+
+AC_MSG_CHECKING(where helpers are installed)
+if test x`basename "$libexecdir"` = "xlibexec"; then
+        AC_MSG_RESULT($libexecdir)
+        CK_LIBEXECDIR="$libexecdir"
+else
+        AC_MSG_RESULT("$libdir/Consolekit, $libdir/policykit-1-gnome, $libdir/gnome-session")
+        CK_LIBEXECDIR="$libdir/ConsoleKit"
+fi
+AC_SUBST(CK_LIBEXECDIR)
+
+dnl ---------------------------------------------------------------------------
 dnl - Configuration file stuff
 dnl ---------------------------------------------------------------------------
 
Index: gdm-3.2.1.1/daemon/Makefile.am
===================================================================
--- gdm-3.2.1.1.orig/daemon/Makefile.am	2011-10-19 17:41:52.000000000 +0200
+++ gdm-3.2.1.1/daemon/Makefile.am	2011-12-17 03:27:51.256987029 +0100
@@ -12,6 +12,7 @@ AM_CPPFLAGS = \
 	-DGDMCONFDIR=\"$(gdmconfdir)\"			\
 	-DLIBDIR=\"$(libdir)\"				\
 	-DLIBEXECDIR=\"$(libexecdir)\"			\
+	-DCK_LIBEXECDIR=\"$(CK_LIBEXECDIR)\"			\
 	-DLOGDIR=\"$(logdir)\"				\
 	-DSBINDIR=\"$(sbindir)\"			\
 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"	\
