Index: gdm-3.14.2/daemon/gdm-session.c
===================================================================
--- gdm-3.14.2.orig/daemon/gdm-session.c	2015-07-09 13:06:02.127447024 +0200
+++ gdm-3.14.2/daemon/gdm-session.c	2015-07-09 13:06:02.123446952 +0200
@@ -1977,6 +1977,7 @@
                                0,
                                service_name);
 
+		gdm_session_select_session_type (self, "x11");
                 gdm_session_authenticate (self, service_name);
                 g_variant_unref (ret);
 
