Index: gdm-3.5.4.2~git20120803~00576a2/daemon/gdm-session.c
===================================================================
--- gdm-3.5.4.2~git20120803~00576a2.orig/daemon/gdm-session.c	2012-08-05 16:30:04.000000000 +0200
+++ gdm-3.5.4.2~git20120803~00576a2/daemon/gdm-session.c	2012-08-05 17:13:59.000000000 +0200
@@ -1903,6 +1903,7 @@
                                0,
                                service_name);
 
+		gdm_session_select_session_type (self, "x11");
                 gdm_session_authenticate (self, service_name);
 
         } else {
