Index: gdm-3.0.0/daemon/gdm-simple-slave.c
===================================================================
--- gdm-3.0.0.orig/daemon/gdm-simple-slave.c	2011-04-04 16:37:16.000000000 +0200
+++ gdm-3.0.0/daemon/gdm-simple-slave.c	2011-04-28 19:40:31.221050273 +0200
@@ -643,7 +643,7 @@ on_session_conversation_started (GdmSess
         } else {
                 g_debug ("GdmSimpleSlave: begin auto login for user '%s'", username);
                 gdm_session_setup_for_user (GDM_SESSION (slave->priv->session),
-                                            "gdm-autologin",
+                                            GDM_PAM_DOMAIN "-autologin",
                                             username);
         }
 
@@ -952,7 +952,7 @@ on_greeter_begin_verification (GdmGreete
 {
         g_debug ("GdmSimpleSlave: begin verification");
         gdm_session_setup (GDM_SESSION (slave->priv->session),
-                           "gdm");
+                           GDM_PAM_DOMAIN);
 }
 
 static void
@@ -965,7 +965,7 @@ on_greeter_begin_auto_login (GdmGreeterS
                 return;
         }
         gdm_session_setup_for_user (GDM_SESSION (slave->priv->session),
-                                    "gdm-autologin",
+                                    GDM_PAM_DOMAIN "-autologin",
                                     username);
 }
 
@@ -979,7 +979,7 @@ on_greeter_begin_verification_for_user (
                 return;
         }
         gdm_session_setup_for_user (GDM_SESSION (slave->priv->session),
-                                    "gdm",
+                                    GDM_PAM_DOMAIN,
                                     username);
 }
 
Index: gdm-3.0.0/configure.ac
===================================================================
--- gdm-3.0.0.orig/configure.ac	2011-04-28 19:07:37.255400047 +0200
+++ gdm-3.0.0/configure.ac	2011-04-28 19:40:31.225050291 +0200
@@ -499,6 +499,21 @@ else
 fi
 AC_SUBST(PAM_PREFIX)
 
+AC_ARG_WITH(pam-domain,
+            AS_HELP_STRING([--with-pam-domain=<domain>],
+                           [specify the naming scheme of pam files]),
+            [if test x$withval != x; then
+               AC_MSG_RESULT("PAM files will be named ${withval} and ${withval}-autologin.")
+             fi])
+
+if test x$withval != x; then
+	GDM_PAM_DOMAIN="$withval"
+else
+	GDM_PAM_DOMAIN=gdm
+fi
+AC_SUBST(GDM_PAM_DOMAIN)
+AC_DEFINE_UNQUOTED([GDM_PAM_DOMAIN], "$GDM_PAM_DOMAIN", [Naming scheme for PAM files])
+
 have_pam=no
 AC_CHECK_LIB(pam, pam_start, have_pam=yes)
 
