--- a/daemon/gdm-simple-slave.c
+++ b/daemon/gdm-simple-slave.c
@@ -593,7 +593,7 @@
         gdm_simple_slave_grant_console_permissions (slave);
 #endif  /* HAVE_LOGINDEVPERM */
 
-        if (g_strcmp0 (service_name, "gdm-autologin") == 0 &&
+        if (g_strcmp0 (service_name, GDM_PAM_DOMAIN"-autologin") == 0 &&
             !gdm_session_client_is_connected (slave->priv->session)) {
                 /* If we're auto logging in then don't wait for the go-ahead from a greeter,
                  * (since there is no greeter) */
@@ -614,7 +614,7 @@
 
         g_debug ("GdmSimpleSlave: session conversation started for service %s", service_name);
 
-        if (g_strcmp0 (service_name, "gdm-autologin") != 0) {
+        if (g_strcmp0 (service_name, GDM_PAM_DOMAIN"-autologin") != 0) {
                 return;
         }
 
@@ -659,7 +659,8 @@
         }
 
         g_debug ("GdmSimpleSlave: Starting automatic login conversation");
-        gdm_session_start_conversation (slave->priv->session, "gdm-autologin");
+        gdm_session_start_conversation (slave->priv->session,
+          GDM_PAM_DOMAIN"-autologin");
 }
 
 static void
--- a/configure.ac
+++ b/configure.ac
@@ -530,6 +530,21 @@
 fi
 AC_SUBST(PAM_PREFIX)
 
+AC_ARG_WITH(pam-domain,
+            AS_HELP_STRING([--with-pam-domain=<domain>],
+                           [specify the naming scheme of pam files]),
+            [if test x$withval != x; then
+               AC_MSG_RESULT("PAM files will be named ${withval} and ${withval}-autologin.")
+             fi])
+
+if test x$withval != x; then
+	GDM_PAM_DOMAIN="$withval"
+else
+	GDM_PAM_DOMAIN=gdm
+fi
+AC_SUBST(GDM_PAM_DOMAIN)
+AC_DEFINE_UNQUOTED([GDM_PAM_DOMAIN], "$GDM_PAM_DOMAIN", [Naming scheme for PAM files])
+
 have_pam=no
 AC_CHECK_LIB(pam, pam_start, have_pam=yes)
 
--- a/daemon/Makefile.am
+++ b/daemon/Makefile.am
@@ -21,6 +21,7 @@
 	-DGDM_SCREENSHOT_DIR=\"$(GDM_SCREENSHOT_DIR)\"		\
 	-DGDM_CACHE_DIR=\""$(localstatedir)/cache/gdm"\"	\
 	-DGDM_SESSION_DEFAULT_PATH=\"$(GDM_SESSION_DEFAULT_PATH)\" \
+	-DGDM_PAM_DOMAIN="\"$(GDM_PAM_DOMAIN)\"" \
 	-DCONSOLEKIT_DIR=\"$(CONSOLEKIT_DIR)\" \
 	$(DISABLE_DEPRECATED_CFLAGS)			\
 	$(DAEMON_CFLAGS)				\
--- a/daemon/gdm-session.c
+++ b/daemon/gdm-session.c
@@ -1322,7 +1322,7 @@
 
         g_debug ("GdmSession: begin auto login for user '%s'", username);
 
-        gdm_session_setup_for_user (self, "gdm-autologin", username);
+        gdm_session_setup_for_user (self, GDM_PAM_DOMAIN"-autologin", username);
 
         return TRUE;
 }
