Description: Load xsplash on login
Bug: https://wiki.ubuntu.com/DesktopTeam/Specs/Karmic/LoginExperience

=== modified file 'data/Init.in'
Index: gdm-2.32.0/data/Init.in
===================================================================
--- gdm-2.32.0.orig/data/Init.in	2010-08-27 16:54:37.000000000 +0200
+++ gdm-2.32.0/data/Init.in	2010-12-07 15:11:07.087501999 +0100
@@ -6,6 +6,11 @@
 PATH="@X_PATH@:$PATH"
 OLD_IFS=$IFS
 
+#if [ -x '/usr/bin/xsplash' ];
+#then
+#        /usr/bin/xsplash --gdm-session --daemon
+#fi
+
 gdmwhich () {
   COMMAND="$1"
   OUTPUT=
Index: gdm-2.32.0/data/PreSession.in
===================================================================
--- gdm-2.32.0.orig/data/PreSession.in	2010-08-27 16:54:37.000000000 +0200
+++ gdm-2.32.0/data/PreSession.in	2010-12-07 15:11:07.087501999 +0100
@@ -7,3 +7,8 @@
 # Note that output goes into the .xsession-errors file for easy debugging
 #
 PATH="@X_PATH@:$PATH"
+
+#if [ -x '/usr/bin/xsplash' ];
+#then
+#        /usr/bin/xsplash --daemon
+#fi
Index: gdm-2.32.0/gui/simple-greeter/gdm-greeter-session.c
===================================================================
--- gdm-2.32.0.orig/gui/simple-greeter/gdm-greeter-session.c	2010-09-15 20:58:11.000000000 +0200
+++ gdm-2.32.0/gui/simple-greeter/gdm-greeter-session.c	2010-12-07 15:11:07.087501999 +0100
@@ -457,6 +457,13 @@
         toggle_panel (session, TRUE);
         toggle_login_window (session, TRUE);
 
+        g_spawn_command_line_async ("dbus-send "
+                                    "--system --type=method_call "
+                                    "--dest=com.ubuntu.BootCurtain "
+                                    "/com/ubuntu/BootCurtain "
+                                    "com.ubuntu.BootCurtain.SignalLoaded string:gdm",
+                                    NULL);
+
         gdm_profile_end (NULL);
 
         return res;
