Description: Load xsplash on login
Bug: https://wiki.ubuntu.com/DesktopTeam/Specs/Karmic/LoginExperience

=== modified file 'data/Init.in'
diff -Nur -x '*.orig' -x '*~' gdm-2.29.5/data/Init.in gdm-2.29.5.new/data/Init.in
--- gdm-2.29.5/data/Init.in	2010-01-13 18:21:31.000000000 +0100
+++ gdm-2.29.5.new/data/Init.in	2010-01-14 16:29:30.705472774 +0100
@@ -6,6 +6,11 @@
 PATH="@X_PATH@:$PATH"
 OLD_IFS=$IFS
 
+#if [ -x '/usr/bin/xsplash' ];
+#then
+#        /usr/bin/xsplash --gdm-session --daemon
+#fi
+
 gdmwhich () {
   COMMAND="$1"
   OUTPUT=
diff -Nur -x '*.orig' -x '*~' gdm-2.29.5/data/PreSession.in gdm-2.29.5.new/data/PreSession.in
--- gdm-2.29.5/data/PreSession.in	2010-01-13 18:21:31.000000000 +0100
+++ gdm-2.29.5.new/data/PreSession.in	2010-01-14 16:29:47.914426562 +0100
@@ -7,3 +7,8 @@
 # Note that output goes into the .xsession-errors file for easy debugging
 #
 PATH="@X_PATH@:$PATH"
+
+#if [ -x '/usr/bin/xsplash' ];
+#then
+#        /usr/bin/xsplash --daemon
+#fi
diff -Nur -x '*.orig' -x '*~' gdm-2.29.5/gui/simple-greeter/gdm-greeter-session.c gdm-2.29.5.new/gui/simple-greeter/gdm-greeter-session.c
--- gdm-2.29.5/gui/simple-greeter/gdm-greeter-session.c	2010-01-12 17:36:37.000000000 +0100
+++ gdm-2.29.5.new/gui/simple-greeter/gdm-greeter-session.c	2010-01-14 16:29:30.705472774 +0100
@@ -439,6 +439,13 @@
         toggle_panel (session, TRUE);
         toggle_login_window (session, TRUE);
 
+        g_spawn_command_line_async ("dbus-send "
+                                    "--system --type=method_call "
+                                    "--dest=com.ubuntu.BootCurtain "
+                                    "/com/ubuntu/BootCurtain "
+                                    "com.ubuntu.BootCurtain.SignalLoaded string:gdm",
+                                    NULL);
+
         gdm_profile_end (NULL);
 
         return res;
