diff -Nur -x '*.orig' -x '*~' gdm-2.27.4/daemon/factory-slave-main.c gdm-2.27.4.new/daemon/factory-slave-main.c
--- gdm-2.27.4/daemon/factory-slave-main.c	2009-06-23 08:05:32.000000000 +0200
+++ gdm-2.27.4.new/daemon/factory-slave-main.c	2009-08-13 20:00:15.000000000 +0200
@@ -212,7 +212,7 @@
         gdm_signal_handler_add (signal_handler, SIGBUS, signal_cb, NULL);
         gdm_signal_handler_add (signal_handler, SIGFPE, signal_cb, NULL);
         gdm_signal_handler_add (signal_handler, SIGHUP, signal_cb, NULL);
-        gdm_signal_handler_add (signal_handler, SIGSEGV, signal_cb, NULL);
+        /*gdm_signal_handler_add (signal_handler, SIGSEGV, signal_cb, NULL);*/
         gdm_signal_handler_add (signal_handler, SIGABRT, signal_cb, NULL);
         gdm_signal_handler_add (signal_handler, SIGUSR1, signal_cb, NULL);
         gdm_signal_handler_add (signal_handler, SIGUSR2, signal_cb, NULL);
diff -Nur -x '*.orig' -x '*~' gdm-2.27.4/daemon/product-slave-main.c gdm-2.27.4.new/daemon/product-slave-main.c
--- gdm-2.27.4/daemon/product-slave-main.c	2009-05-19 17:18:12.000000000 +0200
+++ gdm-2.27.4.new/daemon/product-slave-main.c	2009-08-13 20:00:36.000000000 +0200
@@ -216,7 +216,7 @@
         gdm_signal_handler_add (signal_handler, SIGBUS, signal_cb, NULL);
         gdm_signal_handler_add (signal_handler, SIGFPE, signal_cb, NULL);
         gdm_signal_handler_add (signal_handler, SIGHUP, signal_cb, NULL);
-        gdm_signal_handler_add (signal_handler, SIGSEGV, signal_cb, NULL);
+        /*gdm_signal_handler_add (signal_handler, SIGSEGV, signal_cb, NULL);*/
         gdm_signal_handler_add (signal_handler, SIGABRT, signal_cb, NULL);
         gdm_signal_handler_add (signal_handler, SIGUSR1, signal_cb, NULL);
         gdm_signal_handler_add (signal_handler, SIGUSR2, signal_cb, NULL);
diff -Nur -x '*.orig' -x '*~' gdm-2.27.4/daemon/session-worker-main.c gdm-2.27.4.new/daemon/session-worker-main.c
--- gdm-2.27.4/daemon/session-worker-main.c	2009-06-23 08:05:22.000000000 +0200
+++ gdm-2.27.4.new/daemon/session-worker-main.c	2009-08-13 20:00:19.000000000 +0200
@@ -175,7 +175,7 @@
         gdm_signal_handler_add (signal_handler, SIGBUS, signal_cb, NULL);
         gdm_signal_handler_add (signal_handler, SIGFPE, signal_cb, NULL);
         gdm_signal_handler_add (signal_handler, SIGHUP, signal_cb, NULL);
-        gdm_signal_handler_add (signal_handler, SIGSEGV, signal_cb, NULL);
+        /*gdm_signal_handler_add (signal_handler, SIGSEGV, signal_cb, NULL);*/
         gdm_signal_handler_add (signal_handler, SIGABRT, signal_cb, NULL);
         gdm_signal_handler_add (signal_handler, SIGUSR1, signal_cb, NULL);
 
diff -Nur -x '*.orig' -x '*~' gdm-2.27.4/daemon/simple-slave-main.c gdm-2.27.4.new/daemon/simple-slave-main.c
--- gdm-2.27.4/daemon/simple-slave-main.c	2009-06-23 08:05:25.000000000 +0200
+++ gdm-2.27.4.new/daemon/simple-slave-main.c	2009-08-13 20:00:29.000000000 +0200
@@ -240,7 +240,7 @@
         gdm_signal_handler_add (signal_handler, SIGBUS, signal_cb, NULL);
         gdm_signal_handler_add (signal_handler, SIGFPE, signal_cb, NULL);
         gdm_signal_handler_add (signal_handler, SIGHUP, signal_cb, NULL);
-        gdm_signal_handler_add (signal_handler, SIGSEGV, signal_cb, NULL);
+        /*gdm_signal_handler_add (signal_handler, SIGSEGV, signal_cb, NULL);*/
         gdm_signal_handler_add (signal_handler, SIGABRT, signal_cb, NULL);
         gdm_signal_handler_add (signal_handler, SIGUSR1, signal_cb, NULL);
         gdm_signal_handler_add (signal_handler, SIGUSR2, signal_cb, NULL);
diff -Nur -x '*.orig' -x '*~' gdm-2.27.4/daemon/xdmcp-chooser-slave-main.c gdm-2.27.4.new/daemon/xdmcp-chooser-slave-main.c
--- gdm-2.27.4/daemon/xdmcp-chooser-slave-main.c	2009-06-23 08:05:29.000000000 +0200
+++ gdm-2.27.4.new/daemon/xdmcp-chooser-slave-main.c	2009-08-13 20:00:10.000000000 +0200
@@ -232,7 +232,7 @@
         gdm_signal_handler_add (signal_handler, SIGBUS, signal_cb, NULL);
         gdm_signal_handler_add (signal_handler, SIGFPE, signal_cb, NULL);
         gdm_signal_handler_add (signal_handler, SIGHUP, signal_cb, NULL);
-        gdm_signal_handler_add (signal_handler, SIGSEGV, signal_cb, NULL);
+        /*gdm_signal_handler_add (signal_handler, SIGSEGV, signal_cb, NULL);*/
         gdm_signal_handler_add (signal_handler, SIGABRT, signal_cb, NULL);
         gdm_signal_handler_add (signal_handler, SIGUSR1, signal_cb, NULL);
         gdm_signal_handler_add (signal_handler, SIGUSR2, signal_cb, NULL);
