diff -Nur -x '*.orig' -x '*~' gdm-new-2.25.2/common/gdm-settings-desktop-backend.c gdm-new-2.25.2.new/common/gdm-settings-desktop-backend.c
--- gdm-new-2.25.2/common/gdm-settings-desktop-backend.c	2008-08-26 21:04:00.000000000 +0200
+++ gdm-new-2.25.2.new/common/gdm-settings-desktop-backend.c	2009-03-03 17:39:36.000000000 +0100
@@ -327,7 +327,7 @@
         backend->priv = GDM_SETTINGS_DESKTOP_BACKEND_GET_PRIVATE (backend);
 
         backend->priv->key_file = g_key_file_new ();
-        backend->priv->filename = g_strdup (GDMCONFDIR "/custom.conf");
+        backend->priv->filename = g_strdup (GDMCONFDIR "/gdm.conf-custom");
 
         error = NULL;
         res = g_key_file_load_from_file (backend->priv->key_file,
