Index: gdm-3.0.4/configure.ac
===================================================================
--- gdm-3.0.4.orig/configure.ac	2011-10-14 14:31:15.351512490 +0200
+++ gdm-3.0.4/configure.ac	2011-10-14 14:31:19.227463762 +0200
@@ -1157,7 +1157,10 @@
 # and /usr/X11 since they often symlink to each other, and configure
 # should use the more stable location (the real directory) if possible.
 #
-if test -x /usr/X11/bin/Xserver; then
+if test -n "$X_PATH"; then
+   # assume the user set these variables and don't second-guess them.
+   true
+elif test -x /usr/X11/bin/Xserver; then
    X_PATH="/usr/X11/bin"
    X_SERVER_PATH="/usr/X11/bin"
    X_SERVER="/usr/X11/bin/Xserver"
