daemon: add switch-on-finish property to display

Index: gdm3-3.12.2/daemon/gdm-display.c
===================================================================
--- gdm3-3.12.2.orig/daemon/gdm-display.c	2014-08-14 01:09:48.102831204 +0200
+++ gdm3-3.12.2/daemon/gdm-display.c	2014-08-14 01:10:35.310113921 +0200
@@ -65,6 +65,7 @@ struct GdmDisplayPrivate
         GdmDisplayAccessFile *access_file;
 
         gboolean              is_local;
+        gboolean              switch_on_finish;
         guint                 finish_idle_id;
 
         GdmSlave             *slave;
@@ -89,6 +90,7 @@ enum {
         PROP_X11_COOKIE,
         PROP_X11_AUTHORITY_FILE,
         PROP_IS_LOCAL,
+        PROP_SWITCH_ON_FINISH,
         PROP_SLAVE_TYPE,
         PROP_IS_INITIAL
 };
@@ -135,6 +137,14 @@ gdm_display_get_session_id (GdmDisplay *
         return display->priv->session_id;
 }
 
+gboolean
+gdm_display_get_switch_on_finish (GdmDisplay *display)
+{
+        g_return_val_if_fail (GDM_IS_DISPLAY (display), 0);
+
+        return display->priv->switch_on_finish;
+}
+
 static GdmDisplayAccessFile *
 _create_access_file_for_user (GdmDisplay  *display,
                               const char  *username,
@@ -801,6 +811,13 @@ _gdm_display_set_session_id (GdmDisplay
 }
 
 static void
+_gdm_display_set_switch_on_finish (GdmDisplay   *display,
+                                   gboolean      switch_on_finish)
+{
+        display->priv->switch_on_finish = switch_on_finish;
+}
+
+static void
 _gdm_display_set_remote_hostname (GdmDisplay     *display,
                                   const char     *hostname)
 {
@@ -852,6 +869,16 @@ _gdm_display_set_is_initial (GdmDisplay
         display->priv->is_initial = initial;
 }
 
+void
+gdm_display_set_switch_on_finish (GdmDisplay   *display,
+                                  gboolean      switch_display)
+{
+        g_debug ("GdmDisplay: Setting display %s to %sswitch to login screen on finish",
+                 display->priv->x11_display_name, switch_display? "" : "not ");
+
+        _gdm_display_set_switch_on_finish (display, switch_display);
+}
+
 static void
 gdm_display_set_property (GObject        *object,
                           guint           prop_id,
@@ -890,6 +917,9 @@ gdm_display_set_property (GObject
         case PROP_IS_LOCAL:
                 _gdm_display_set_is_local (self, g_value_get_boolean (value));
                 break;
+        case PROP_SWITCH_ON_FINISH:
+                _gdm_display_set_switch_on_finish (self, g_value_get_boolean (value));
+                break;
         case PROP_SLAVE_TYPE:
                 _gdm_display_set_slave_type (self, g_value_get_gtype (value));
                 break;
@@ -944,6 +974,9 @@ gdm_display_get_property (GObject
         case PROP_IS_LOCAL:
                 g_value_set_boolean (value, self->priv->is_local);
                 break;
+        case PROP_SWITCH_ON_FINISH:
+                g_value_set_boolean (value, self->priv->switch_on_finish);
+                break;
         case PROP_SLAVE_TYPE:
                 g_value_set_gtype (value, self->priv->slave_type);
                 break;
@@ -1351,6 +1384,14 @@ gdm_display_class_init (GdmDisplayClass
                                                                NULL,
                                                                NULL,
                                                                TRUE,
+                                                               G_PARAM_READWRITE | G_PARAM_CONSTRUCT));
+
+        g_object_class_install_property (object_class,
+                                         PROP_SWITCH_ON_FINISH,
+                                         g_param_spec_boolean ("switch-on-finish",
+                                                               NULL,
+                                                               NULL,
+                                                               TRUE,
                                                                G_PARAM_READWRITE | G_PARAM_CONSTRUCT));
 
         g_object_class_install_property (object_class,
Index: gdm3-3.12.2/daemon/gdm-display.h
===================================================================
--- gdm3-3.12.2.orig/daemon/gdm-display.h	2014-08-14 01:09:48.102831204 +0200
+++ gdm3-3.12.2/daemon/gdm-display.h	2014-08-14 01:09:48.098831265 +0200
@@ -136,6 +136,9 @@ gboolean            gdm_display_get_time
 gboolean            gdm_display_is_initial                     (GdmDisplay  *display,
                                                                 gboolean    *initial,
                                                                 GError     **error);
+gboolean            gdm_display_get_switch_on_finish           (GdmDisplay  *display);
+void                gdm_display_set_switch_on_finish           (GdmDisplay  *display,
+                                                                gboolean     switch_display);
 
 gboolean            gdm_display_get_x11_cookie                 (GdmDisplay  *display,
                                                                 const char **x11_cookie,
Index: gdm3-3.12.2/daemon/gdm-transient-display.c
===================================================================
--- gdm3-3.12.2.orig/daemon/gdm-transient-display.c	2014-08-14 01:09:48.102831204 +0200
+++ gdm3-3.12.2/daemon/gdm-transient-display.c	2014-08-14 01:09:52.000000000 +0200
@@ -188,6 +188,7 @@ gdm_transient_display_new (int display_n
         object = g_object_new (GDM_TYPE_TRANSIENT_DISPLAY,
                                "x11-display-number", display_number,
                                "x11-display-name", x11_display,
+                               "switch-on-finish", TRUE,
                                NULL);
         g_free (x11_display);
 
