diff -Nur -x '*.orig' -x '*~' gdm-2.26.1/configure.ac gdm-2.26.1.new/configure.ac
--- gdm-2.26.1/configure.ac	2009-06-30 23:23:03.000000000 +0200
+++ gdm-2.26.1.new/configure.ac	2009-06-30 23:23:04.000000000 +0200
@@ -1136,9 +1136,9 @@
 else
    # what to do, what to do, this is wrong, but this just sets the
    # defaults, perhaps this user is cross compiling or some such
-   X_PATH="/usr/bin/X11:/usr/X11R6/bin:/opt/X11R6/bin"
-   X_SERVER_PATH="/usr/X11R6/bin"
-   X_SERVER="/usr/X11R6/bin/X"
+   X_PATH="/usr/bin"
+   X_SERVER_PATH="/usr/bin"
+   X_SERVER="/usr/bin/X"
    X_CONFIG_OPTIONS="-br -audit 0"
 fi
 
