diff -Nur -x '*.orig' -x '*~' gdm-2.26.1/configure gdm-2.26.1.new/configure
--- gdm-2.26.1/configure	2009-04-14 16:21:25.000000000 +0200
+++ gdm-2.26.1.new/configure	2009-06-30 23:23:07.000000000 +0200
@@ -31186,32 +31186,32 @@
    X_PATH="/usr/X11/bin"
    X_SERVER_PATH="/usr/X11/bin"
    X_SERVER="/usr/X11/bin/Xserver"
-   X_CONFIG_OPTIONS="-audit 0"
+   X_CONFIG_OPTIONS="-br -audit 0"
 elif test ! -h /usr/X11R6 -a -x /usr/X11R6/bin/X; then
    X_PATH="/usr/X11R6/bin"
    X_SERVER_PATH="/usr/X11R6/bin"
    X_SERVER="/usr/X11R6/bin/X"
-   X_CONFIG_OPTIONS="-audit 0"
+   X_CONFIG_OPTIONS="-br -audit 0"
 elif test ! -h /usr/X11 -a -x /usr/X11/bin/X; then
    X_PATH="/usr/X11/bin"
    X_SERVER_PATH="/usr/X11/bin"
    X_SERVER="/usr/X11/bin/X"
-   X_CONFIG_OPTIONS="-audit 0"
+   X_CONFIG_OPTIONS="-br -audit 0"
 elif test -x /usr/X11R6/bin/X; then
    X_PATH="/usr/X11R6/bin"
    X_SERVER_PATH="/usr/X11R6/bin"
    X_SERVER="/usr/X11R6/bin/X"
-   X_CONFIG_OPTIONS="-audit 0"
+   X_CONFIG_OPTIONS="-br -audit 0"
 elif test -x /usr/bin/Xorg; then
    X_PATH="/usr/bin"
    X_SERVER_PATH="/usr/bin"
    X_SERVER="/usr/bin/Xorg"
-   X_CONFIG_OPTIONS="-audit 0"
+   X_CONFIG_OPTIONS="-br -audit 0"
 elif test -x /usr/X11/bin/X; then
    X_PATH="/usr/X11/bin"
    X_SERVER_PATH="/usr/X11/bin"
    X_SERVER="/usr/X11/bin/X"
-   X_CONFIG_OPTIONS="-audit 0"
+   X_CONFIG_OPTIONS="-br -audit 0"
 elif test -x /usr/openwin/bin/Xsun; then
    # Do not add /usr/openwin/bin here because on Solaris you need
    # /usr/openwin/bin in your PATH even if you are using the Xorg
@@ -31219,24 +31219,24 @@
    X_PATH="/usr/openwin/bin"
    X_SERVER_PATH="/usr/openwin/bin"
    X_SERVER="/usr/openwin/bin/Xsun"
-   X_CONFIG_OPTIONS="-audit 0 -nobanner"
+   X_CONFIG_OPTIONS="-br -audit 0 -nobanner"
 elif test -x /opt/X11R6/bin/X; then
    X_PATH="/opt/X11R6/bin"
    X_SERVER_PATH="/opt/X11R6/bin"
    X_SERVER="/opt/X11R6/bin/X"
-   X_CONFIG_OPTIONS="-audit 0"
+   X_CONFIG_OPTIONS="-br -audit 0"
 elif test -x /usr/bin/X; then
    X_PATH="/usr/bin"
    X_SERVER_PATH="/usr/bin"
    X_SERVER="/usr/bin/X"
-   X_CONFIG_OPTIONS="-audit 0"
+   X_CONFIG_OPTIONS="-br -audit 0"
 else
    # what to do, what to do, this is wrong, but this just sets the
    # defaults, perhaps this user is cross compiling or some such
-   X_PATH="/usr/bin/X11:/usr/X11R6/bin:/opt/X11R6/bin"
-   X_SERVER_PATH="/usr/X11R6/bin"
-   X_SERVER="/usr/X11R6/bin/X"
-   X_CONFIG_OPTIONS="-audit 0"
+   X_PATH="/usr/bin"
+   X_SERVER_PATH="/usr/bin"
+   X_SERVER="/usr/bin/X"
+   X_CONFIG_OPTIONS="-br -audit 0"
 fi
 
 
@@ -31246,7 +31246,7 @@
 X_XNEST_UNSCALED_FONTPATH="true"
 if test -x $X_SERVER_PATH/Xephyr; then
     X_XNEST_CMD="$X_SERVER_PATH/Xephyr"
-    X_XNEST_CONFIG_OPTIONS="-audit 0"
+    X_XNEST_CONFIG_OPTIONS="-br -audit 0"
 else
     if test -x /usr/openwin/bin/Xnest; then
         # If on Solaris, Xnest is only shipped with the Xsun Xserver, so
@@ -31258,7 +31258,7 @@
     else
         if test -x $X_SERVER_PATH/Xnest; then
             X_XNEST_CMD="$X_SERVER_PATH/Xnest"
-            X_XNEST_CONFIG_OPTIONS="-audit 0 -name Xnest"
+            X_XNEST_CONFIG_OPTIONS="-br -audit 0 -name Xnest"
         fi
     fi
 fi
diff -Nur -x '*.orig' -x '*~' gdm-2.26.1/configure.ac gdm-2.26.1.new/configure.ac
--- gdm-2.26.1/configure.ac	2009-06-30 23:23:03.000000000 +0200
+++ gdm-2.26.1.new/configure.ac	2009-06-30 23:23:04.000000000 +0200
@@ -1136,9 +1136,9 @@
 else
    # what to do, what to do, this is wrong, but this just sets the
    # defaults, perhaps this user is cross compiling or some such
-   X_PATH="/usr/bin/X11:/usr/X11R6/bin:/opt/X11R6/bin"
-   X_SERVER_PATH="/usr/X11R6/bin"
-   X_SERVER="/usr/X11R6/bin/X"
+   X_PATH="/usr/bin"
+   X_SERVER_PATH="/usr/bin"
+   X_SERVER="/usr/bin/X"
    X_CONFIG_OPTIONS="-br -audit 0"
 fi
 
