GNOME #586777

Index: gdm-2.30.5/common/gdm-settings-keys.h
===================================================================
--- gdm-2.30.5.orig/common/gdm-settings-keys.h	2010-09-11 11:12:09.144142815 +0200
+++ gdm-2.30.5/common/gdm-settings-keys.h	2010-09-11 11:13:37.816147498 +0200
@@ -33,6 +33,8 @@ G_BEGIN_DECLS
 #define GDM_KEY_TIMED_LOGIN_USER "daemon/TimedLogin"
 #define GDM_KEY_TIMED_LOGIN_DELAY "daemon/TimedLoginDelay"
 #define GDM_KEY_FIRST_VT "daemon/FirstVT"
+#define GDM_KEY_LOCAL_XSERVER_COMMAND "daemon/LocalXserverCommand"
+#define GDM_KEY_NESTED_XSERVER_COMMAND "daemon/NestedXserverCommand"
 
 #define GDM_KEY_DEBUG "debug/Enable"
 
Index: gdm-2.30.5/daemon/gdm-server.c
===================================================================
--- gdm-2.30.5.orig/daemon/gdm-server.c	2010-09-11 11:13:07.040641299 +0200
+++ gdm-2.30.5/daemon/gdm-server.c	2010-09-11 11:13:55.032642229 +0200
@@ -104,6 +104,7 @@ enum {
         PROP_PARENT_AUTH_FILE,
         PROP_CHOSEN_HOSTNAME,
         PROP_COMMAND,
+        PROP_NESTED_COMMAND,
         PROP_PRIORITY,
         PROP_USER_NAME,
         PROP_SESSION_ARGS,
@@ -1028,6 +1029,22 @@ _gdm_server_set_disable_tcp (GdmServer
 }
 
 static void
+_gdm_server_set_command (GdmServer  *server,
+                         const char *command)
+{
+        g_free (server->priv->command);
+        server->priv->command = g_strdup (command);
+}
+
+static void
+_gdm_server_set_nested_command (GdmServer  *server,
+                                const char *command)
+{
+        g_free (server->priv->nested_command);
+        server->priv->nested_command = g_strdup (command);
+}
+
+static void
 gdm_server_set_property (GObject      *object,
                          guint         prop_id,
                          const GValue *value,
@@ -1062,6 +1079,12 @@ gdm_server_set_property (GObject      *o
         case PROP_IS_NESTED:
                 _gdm_server_set_is_nested (self, g_value_get_boolean (value));
                 break;
+        case PROP_COMMAND:
+                _gdm_server_set_command (self, g_value_get_string (value));
+                break;
+        case PROP_NESTED_COMMAND:
+                _gdm_server_set_nested_command (self, g_value_get_string (value));
+                break;
         default:
                 G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
                 break;
@@ -1107,6 +1130,12 @@ gdm_server_get_property (GObject    *obj
         case PROP_IS_NESTED:
                 g_value_set_boolean (value, self->priv->is_nested);
                 break;
+        case PROP_COMMAND:
+                g_value_set_string (value, self->priv->command);
+                break;
+        case PROP_NESTED_COMMAND:
+                g_value_set_string (value, self->priv->nested_command);
+                break;
         default:
                 G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
                 break;
@@ -1208,6 +1237,20 @@ gdm_server_class_init (GdmServerClass *k
                                                                TRUE,
                                                                G_PARAM_READWRITE | G_PARAM_CONSTRUCT));
         g_object_class_install_property (object_class,
+                                         PROP_COMMAND,
+                                         g_param_spec_string ("command",
+                                                              "Server startup command",
+                                                              "Server startup command",
+                                                              NULL,
+                                                              G_PARAM_READWRITE | G_PARAM_CONSTRUCT));
+        g_object_class_install_property (object_class,
+                                         PROP_NESTED_COMMAND,
+                                         g_param_spec_string ("nested-command",
+                                                              "Nested server startup command",
+                                                              "Nested server startup command",
+                                                              NULL,
+                                                              G_PARAM_READWRITE | G_PARAM_CONSTRUCT));
+        g_object_class_install_property (object_class,
                                          PROP_IS_PARENTED,
                                          g_param_spec_boolean ("is-parented",
                                                                NULL,
@@ -1245,8 +1288,6 @@ gdm_server_init (GdmServer *server)
         server->priv = GDM_SERVER_GET_PRIVATE (server);
 
         server->priv->pid = -1;
-        server->priv->command = g_strdup (X_SERVER " -br -verbose");
-        server->priv->nested_command = g_strdup (X_XNEST_CMD " " X_XNEST_CONFIG_OPTIONS " -br")
         server->priv->log_dir = g_strdup (LOGDIR);
         server->priv->is_ready = FALSE;
 
@@ -1291,6 +1332,8 @@ gdm_server_new (const char *display_name
         GObject *object;
 
         object = g_object_new (GDM_TYPE_SERVER,
+                               "command", X_SERVER " -br -verbose " X_CONFIG_OPTIONS,
+                               "nested-command", X_XNEST_CMD " -br " X_XNEST_CONFIG_OPTIONS,
                                "display-name", display_name,
                                "auth-file", auth_file,
                                NULL);
Index: gdm-2.30.5/daemon/gdm-simple-slave.c
===================================================================
--- gdm-2.30.5.orig/daemon/gdm-simple-slave.c	2010-09-11 11:13:37.784154115 +0200
+++ gdm-2.30.5/daemon/gdm-simple-slave.c	2010-09-11 11:13:37.816147498 +0200
@@ -1162,6 +1162,7 @@ gdm_simple_slave_run (GdmSimpleSlave *sl
         if (display_is_local) {
                 gboolean res;
                 gboolean disable_tcp;
+                char    *server_command;
 
                 slave->priv->server = gdm_server_new (display_name, auth_file);
 
@@ -1183,6 +1184,28 @@ gdm_simple_slave_run (GdmSimpleSlave *sl
                                       NULL);
                 }
 
+                server_command = NULL;
+                if (gdm_settings_client_get_string (GDM_KEY_LOCAL_XSERVER_COMMAND,
+                                                    &server_command) &&
+                    server_command != NULL &&
+                    server_command [0] != '\0') {
+                        g_object_set (slave->priv->server,
+                                      "command", server_command,
+                                      NULL);
+                        g_free (server_command);
+                }
+
+                server_command = NULL;
+                if (gdm_settings_client_get_string (GDM_KEY_NESTED_XSERVER_COMMAND,
+                                                    &server_command) &&
+                    server_command != NULL &&
+                    server_command [0] != '\0') {
+                        g_object_set (slave->priv->server,
+                                      "nested-command", server_command,
+                                      NULL);
+                        g_free (server_command);
+                }
+
                 g_signal_connect (slave->priv->server,
                                   "exited",
                                   G_CALLBACK (on_server_exited),
Index: gdm-2.30.5/data/gdm.schemas.in.in
===================================================================
--- gdm-2.30.5.orig/data/gdm.schemas.in.in	2010-09-11 11:12:09.100142319 +0200
+++ gdm-2.30.5/data/gdm.schemas.in.in	2010-09-11 11:13:37.816147498 +0200
@@ -23,6 +23,16 @@
       <default>@GDM_GROUPNAME@</default>
     </schema>
     <schema>
+      <key>daemon/LocalXserverCommand</key>
+      <signature>s</signature>
+      <default>@X_SERVER@ -br -verbose @X_CONFIG_OPTIONS@</default>
+    </schema>
+    <schema>
+      <key>daemon/NestedXserverCommand</key>
+      <signature>s</signature>
+      <default>@X_XNEST_CMD@ -br @X_XNEST_CONFIG_OPTIONS@</default>
+    </schema>
+    <schema>
       <key>daemon/AutomaticLoginEnable</key>
       <signature>b</signature>
       <default>false</default>
