Index: gdm-2.30.2/daemon/gdm-server.c
===================================================================
--- gdm-2.30.2.orig/daemon/gdm-server.c	2010-05-08 10:37:27.353612651 +0200
+++ gdm-2.30.2/daemon/gdm-server.c	2010-05-08 10:37:50.737110527 +0200
@@ -1133,7 +1133,7 @@ gdm_server_init (GdmServer *server)
         server->priv = GDM_SERVER_GET_PRIVATE (server);
 
         server->priv->pid = -1;
-        server->priv->command = g_strdup (X_SERVER " -br -verbose");
+        server->priv->command = g_strdup (X_SERVER " -br -verbose " X_CONFIG_OPTIONS);
         server->priv->log_dir = g_strdup (LOGDIR);
 
         add_ready_handler (server);
