diff -Nur -x '*.orig' -x '*~' gdm/data/Init.in gdm.new/data/Init.in
--- gdm/data/Init.in	2009-09-14 20:42:01.000000000 +0100
+++ gdm.new/data/Init.in	2009-09-14 20:44:40.000000000 +0100
@@ -11,6 +11,8 @@
         /usr/bin/xsplash --gdm-session &
 fi
 
+initctl -q emit login-session-start DISPLAY_MANAGER=gdm
+
 gdmwhich () {
   COMMAND="$1"
   OUTPUT=
diff -Nur -x '*.orig' -x '*~' gdm/data/PreSession.in gdm.new/data/PreSession.in
--- gdm/data/PreSession.in	2009-09-14 20:42:01.000000000 +0100
+++ gdm.new/data/PreSession.in	2009-09-14 20:44:53.000000000 +0100
@@ -13,3 +13,4 @@
         /usr/bin/xsplash &
 fi
 
+initctl -q emit desktop-session-start DISPLAY_MANAGER=gdm
