diff -Nur -x '*.orig' -x '*~' gdm-2.28.0/configure.ac gdm-2.28.0.new/configure.ac
--- gdm-2.28.0/configure.ac	2009-09-22 22:25:14.792750682 +0100
+++ gdm-2.28.0.new/configure.ac	2009-09-22 22:25:15.732749468 +0100
@@ -52,6 +52,7 @@
 LIBCANBERRA_GTK_REQUIRED_VERSION=0.4
 #FONTCONFIG_REQUIRED_VERSION=2.6.0
 FONTCONFIG_REQUIRED_VERSION=2.5.0
+DEVKIT_POWER_REQUIRED=008
 
 EXTRA_COMPILE_WARNINGS(yes)
 
@@ -122,6 +123,7 @@
         libglade-2.0 >= $LIBGLADE_REQUIRED_VERSION
         gconf-2.0 >= $GCONF_REQUIRED_VERSION
 	fontconfig >= $FONTCONFIG_REQUIRED_VERSION
+	devkit-power-gobject >= $DEVKIT_POWER_REQUIRED
 )
 AC_SUBST(SIMPLE_GREETER_CFLAGS)
 AC_SUBST(SIMPLE_GREETER_LIBS)
diff -Nur -x '*.orig' -x '*~' gdm-2.28.0/gui/simple-greeter/gdm-greeter-login-window.c gdm-2.28.0.new/gui/simple-greeter/gdm-greeter-login-window.c
--- gdm-2.28.0/gui/simple-greeter/gdm-greeter-login-window.c	2009-09-22 22:25:14.560748383 +0100
+++ gdm-2.28.0.new/gui/simple-greeter/gdm-greeter-login-window.c	2009-09-22 22:25:31.948750678 +0100
@@ -44,6 +44,8 @@
 #include <glib/gstdio.h>
 #include <glib-object.h>
 
+#include <devkit-power-gobject/devicekit-power.h>
+
 #include <gdk/gdkkeysyms.h>
 #include <gdk/gdkx.h>
 #include <X11/XKBlib.h>
@@ -79,10 +81,6 @@
 #define CK_SEAT_INTERFACE    "org.freedesktop.ConsoleKit.Seat"
 #define CK_SESSION_INTERFACE "org.freedesktop.ConsoleKit.Session"
 
-#define GPM_DBUS_NAME      "org.freedesktop.PowerManagement"
-#define GPM_DBUS_PATH      "/org/freedesktop/PowerManagement"
-#define GPM_DBUS_INTERFACE "org.freedesktop.PowerManagement"
-
 #define GLADE_XML_FILE       "gdm-greeter-login-window.glade"
 
 #define KEY_GREETER_DIR             "/apps/gdm/simple-greeter"
@@ -129,6 +127,8 @@
 
         guint            login_button_handler_id;
         guint            start_session_handler_id;
+        
+        DkpClient       *dkp_client;
 };
 
 enum {
@@ -477,46 +477,6 @@
         }
 }
 
-static gboolean
-can_suspend (GdmGreeterLoginWindow *login_window)
-{
-        DBusGConnection *connection;
-        DBusGProxy      *proxy;
-        GError          *error;
-        gboolean         ret;
-        gboolean         res;
-
-        error = NULL;
-        connection = dbus_g_bus_get (DBUS_BUS_SESSION, &error);
-        if (error != NULL) {
-                g_warning ("Couldn't connect to power manager: %s", error->message);
-                g_error_free (error);
-                return FALSE;
-        }
-        proxy = dbus_g_proxy_new_for_name (connection,
-                                           GPM_DBUS_NAME,
-                                           GPM_DBUS_PATH,
-                                           GPM_DBUS_INTERFACE);
-        ret = FALSE;
-
-        res = dbus_g_proxy_call (proxy, "CanSuspend",
-                                 &error,
-                                 G_TYPE_INVALID,
-                                 G_TYPE_BOOLEAN,
-                                 &ret,
-                                 G_TYPE_INVALID);
-        if (! res) {
-                if (error != NULL) {
-                        g_warning ("Could not ask power manager if user can suspend: %s",
-                                   error->message);
-                        g_error_free (error);
-                }
-                ret = FALSE;
-        }
-
-        return ret;
-}
-
 static void
 remove_sensitize_power_buttons_timeout (GdmGreeterLoginWindow *login_window)
 {
@@ -534,6 +494,7 @@
                 sensitize_widget (login_window, "shutdown-button", TRUE);
                 sensitize_widget (login_window, "restart-button", TRUE);
                 sensitize_widget (login_window, "suspend-button", TRUE);
+                sensitize_widget (login_window, "hibernate-button", TRUE);
                 sensitize_widget (login_window, "disconnect-button", TRUE);
                 break;
         case MODE_AUTHENTICATION:
@@ -564,9 +525,11 @@
         GtkWidget  *box;
         gboolean    show_restart_buttons;
         gboolean    show_suspend_button;
+        gboolean    show_hibernate_button;
 
         show_restart_buttons = get_show_restart_buttons (login_window);
-        show_suspend_button = can_suspend (login_window);
+        show_suspend_button = dkp_client_can_suspend (login_window->priv->dkp_client);
+        show_hibernate_button = dkp_client_can_hibernate (login_window->priv->dkp_client);
 
         /* we want to run this even if we're supposed to
            be in the mode already so that we reset everything
@@ -589,6 +552,8 @@
                              login_window->priv->display_is_local && show_restart_buttons);
                 show_widget (login_window, "suspend-button",
                              login_window->priv->display_is_local && show_restart_buttons && show_suspend_button);
+                show_widget (login_window, "hibernate-button",
+                             login_window->priv->display_is_local && show_restart_buttons && show_hibernate_button);
                 show_widget (login_window, "disconnect-button",
                              ! login_window->priv->display_is_local);
 
@@ -598,6 +563,7 @@
                 sensitize_widget (login_window, "shutdown-button", FALSE);
                 sensitize_widget (login_window, "restart-button", FALSE);
                 sensitize_widget (login_window, "suspend-button", FALSE);
+                sensitize_widget (login_window, "hibernate-button", FALSE);
                 sensitize_widget (login_window, "disconnect-button", FALSE);
 
                 default_name = NULL;
@@ -607,6 +573,7 @@
                 show_widget (login_window, "shutdown-button", FALSE);
                 show_widget (login_window, "restart-button", FALSE);
                 show_widget (login_window, "suspend-button", FALSE);
+                show_widget (login_window, "hibernate-button", FALSE);
                 show_widget (login_window, "disconnect-button", FALSE);
                 default_name = "log-in-button";
                 break;
@@ -653,41 +620,6 @@
 }
 
 static void
-do_suspend (GdmGreeterLoginWindow *login_window)
-{
-        GError          *error;
-        DBusGConnection *connection;
-        DBusGProxy      *proxy;
-
-        g_debug ("GdmGreeterLoginWindow: Suspend button clicked");
-
-        error = NULL;
-        connection = dbus_g_bus_get (DBUS_BUS_SESSION, &error);
-        if (error != NULL) {
-                g_warning ("Couldn't suspend: %s", error->message);
-                g_error_free (error);
-                return;
-        }
-        proxy = dbus_g_proxy_new_for_name (connection,
-                                           GPM_DBUS_NAME,
-                                           GPM_DBUS_PATH,
-                                           GPM_DBUS_INTERFACE);
-        error = NULL;
-        dbus_g_proxy_call (proxy,
-                           "Suspend",
-                           &error,
-                           G_TYPE_INVALID,
-                           G_TYPE_INVALID);
-        if (error != NULL) {
-                g_warning ("Couldn't suspend: %s", error->message);
-                g_error_free (error);
-                return;
-        }
-
-        g_object_unref (proxy);
-}
-
-static void
 delete_entry_text (GtkWidget *entry)
 {
         const char *typed_text;
@@ -1004,9 +936,31 @@
 suspend_button_clicked (GtkButton             *button,
                         GdmGreeterLoginWindow *login_window)
 {
-        do_suspend (login_window);
+        gboolean ret;
+        GError *error;
+        
+        g_debug ("GdmGreeterLoginWindow: Suspend button clicked");
+        ret = dkp_client_suspend (login_window->priv->dkp_client, &error);
+        if (!ret) {
+                g_warning ("Suspend failure: %s", error->message);
+                g_error_free (error);
+        }
 }
 
+static void
+hibernate_button_clicked (GtkButton             *button,
+                          GdmGreeterLoginWindow *login_window)
+{
+        gboolean ret;
+        GError *error;
+        
+        g_debug ("GdmGreeterLoginWindow: Hibernate button clicked");
+        ret = dkp_client_hibernate (login_window->priv->dkp_client, &error);
+        if (!ret) {
+                g_warning ("Hibernate failure: %s", error->message);
+                g_error_free (error);
+        }
+}
 
 static void
 cancel_button_clicked (GtkButton             *button,
@@ -1437,6 +1391,9 @@
 
         button = glade_xml_get_widget (login_window->priv->xml, "suspend-button");
         g_signal_connect (button, "clicked", G_CALLBACK (suspend_button_clicked), login_window);
+ 
+        button = glade_xml_get_widget (login_window->priv->xml, "hibernate-button");
+        g_signal_connect (button, "clicked", G_CALLBACK (hibernate_button_clicked), login_window);
 
         button = glade_xml_get_widget (login_window->priv->xml, "cancel-button");
         g_signal_connect (button, "clicked", G_CALLBACK (cancel_button_clicked), login_window);
@@ -1796,6 +1753,8 @@
                                                                   login_window,
                                                                   NULL,
                                                                   NULL);
+                                                                  
+        login_window->priv->dkp_client = dkp_client_new ();
         gdm_profile_end (NULL);
 }
 
@@ -1814,6 +1773,10 @@
         if (login_window->priv->client != NULL) {
                 g_object_unref (login_window->priv->client);
         }
+        
+        if (login_window->priv->dkp_client != NULL) {
+                g_object_unref (login_window->priv->dkp_client);
+        }
 
         remove_sensitize_power_buttons_timeout (login_window);
 
diff -Nur -x '*.orig' -x '*~' gdm-2.28.0/gui/simple-greeter/gdm-greeter-login-window.glade gdm-2.28.0.new/gui/simple-greeter/gdm-greeter-login-window.glade
--- gdm-2.28.0/gui/simple-greeter/gdm-greeter-login-window.glade	2009-09-21 21:05:27.000000000 +0100
+++ gdm-2.28.0.new/gui/simple-greeter/gdm-greeter-login-window.glade	2009-09-22 22:25:15.736747287 +0100
@@ -409,6 +409,67 @@
 		  </child>
 
 		  <child>
+		    <widget class="GtkButton" id="hibernate-button">
+		      <property name="visible">True</property>
+		      <property name="can_default">True</property>
+		      <property name="has_default">True</property>
+		      <property name="can_focus">True</property>
+		      <property name="relief">GTK_RELIEF_NORMAL</property>
+		      <property name="focus_on_click">True</property>
+
+		      <child>
+			<widget class="GtkHBox" id="hbox6">
+			  <property name="visible">True</property>
+			  <property name="homogeneous">False</property>
+			  <property name="spacing">0</property>
+
+			  <child>
+			    <widget class="GtkImage" id="image10">
+			      <property name="visible">True</property>
+			      <property name="icon_name">drive-harddisk</property>
+			      <property name="pixel_size">16</property>
+			      <property name="xalign">0.5</property>
+			      <property name="yalign">0.5</property>
+			      <property name="xpad">0</property>
+			      <property name="ypad">0</property>
+			    </widget>
+			    <packing>
+			      <property name="padding">0</property>
+			      <property name="expand">True</property>
+			      <property name="fill">True</property>
+			    </packing>
+			  </child>
+
+			  <child>
+			    <widget class="GtkLabel" id="label5">
+			      <property name="visible">True</property>
+			      <property name="label" translatable="yes">Hibernate</property>
+			      <property name="use_underline">False</property>
+			      <property name="use_markup">False</property>
+			      <property name="justify">GTK_JUSTIFY_LEFT</property>
+			      <property name="wrap">False</property>
+			      <property name="selectable">False</property>
+			      <property name="xalign">0</property>
+			      <property name="yalign">0.5</property>
+			      <property name="xpad">0</property>
+			      <property name="ypad">0</property>
+			      <property name="ellipsize">PANGO_ELLIPSIZE_NONE</property>
+			      <property name="width_chars">-1</property>
+			      <property name="single_line_mode">False</property>
+			      <property name="angle">0</property>
+			    </widget>
+			    <packing>
+			      <property name="padding">0</property>
+			      <property name="expand">True</property>
+			      <property name="fill">True</property>
+			    </packing>
+			  </child>
+			</widget>
+		      </child>
+		    </widget>
+		  </child>
+
+		  <child>
 		    <widget class="GtkButton" id="restart-button">
 		      <property name="visible">True</property>
 		      <property name="can_default">True</property>
diff -Nur -x '*.orig' -x '*~' gdm-2.28.0/gui/simple-greeter/Makefile.am gdm-2.28.0.new/gui/simple-greeter/Makefile.am
--- gdm-2.28.0/gui/simple-greeter/Makefile.am	2009-09-22 22:25:14.720746074 +0100
+++ gdm-2.28.0.new/gui/simple-greeter/Makefile.am	2009-09-22 22:25:15.736747287 +0100
@@ -19,6 +19,7 @@
 	-DSBINDIR=\""$(sbindir)"\"		 	\
 	-DGDM_CACHE_DIR=\""$(localstatedir)/cache/gdm"\"	\
 	-DAT_SPI_REGISTRYD_DIR="\"$(AT_SPI_REGISTRYD_DIR)\""	\
+	-DI_KNOW_THE_DEVICEKIT_POWER_API_IS_SUBJECT_TO_CHANGE	\
 	$(DISABLE_DEPRECATED_CFLAGS)			\
 	$(GTK_CFLAGS)					\
 	$(SIMPLE_GREETER_CFLAGS)			\
