Index: gdm-2.30.2/gui/simple-greeter/gdm-user-chooser-widget.c
===================================================================
--- gdm-2.30.2.orig/gui/simple-greeter/gdm-user-chooser-widget.c	2010-06-12 10:38:48.533310336 +0200
+++ gdm-2.30.2/gui/simple-greeter/gdm-user-chooser-widget.c	2010-06-12 10:40:16.905305863 +0200
@@ -140,13 +140,8 @@ update_other_user_visibility (GdmUserCho
                 return;
         }
 
-        number_of_users = gdm_chooser_widget_get_number_of_items (GDM_CHOOSER_WIDGET (widget));
-
-        /* we hide the Other user if it's the last one, and we show it
-         * if there's another user */
-        if (number_of_users == 1 && widget->priv->has_user_other) {
-                remove_user_other (widget);
-        } if (number_of_users >= 1 && !widget->priv->has_user_other) {
+        /* Always show the Other user if requested */
+        if (!widget->priv->has_user_other) {
                 add_user_other (widget);
         }
 }
