Index: gdm-2.30.5/gui/simple-greeter/gdm-user-chooser-widget.c
===================================================================
--- gdm-2.30.5.orig/gui/simple-greeter/gdm-user-chooser-widget.c	2010-08-11 19:40:07.000000000 +0200
+++ gdm-2.30.5/gui/simple-greeter/gdm-user-chooser-widget.c	2010-09-05 19:48:56.964871434 +0200
@@ -150,13 +150,8 @@ update_other_user_visibility (GdmUserCho
                 goto out;
         }
 
-        number_of_users = gdm_chooser_widget_get_number_of_items (GDM_CHOOSER_WIDGET (widget));
-
-        /* we hide the Other user if it's the last one, and we show it
-         * if there's another user */
-        if (number_of_users == 1 && widget->priv->has_user_other) {
-                remove_user_other (widget);
-        } if (number_of_users >= 1 && !widget->priv->has_user_other) {
+        /* Always show the Other user if requested */
+        if (!widget->priv->has_user_other) {
                 add_user_other (widget);
         }
 
