Index: gdm-2.32.0/data/gconf.path
===================================================================
--- gdm-2.32.0.orig/data/gconf.path	2010-08-27 23:45:29.000000000 +0200
+++ gdm-2.32.0/data/gconf.path	2010-12-07 15:20:49.987502001 +0100
@@ -10,3 +10,8 @@
 # Set per-seat configuration directory
 xml:readwrite:$(HOME)/$(ENV_GDM_SEAT_ID)/.gconf
 
+# user values
+xml:readwrite:$(HOME)/.gconf
+
+# distribution default values
+xml:readwrite:$(HOME)/.gconf.defaults
