diff -Nur -x '*.orig' -x '*~' gdm-2.29.92/data/gconf.path gdm-2.29.92.new/data/gconf.path
--- gdm-2.29.92/data/gconf.path	2010-03-08 22:53:57.000000000 +0100
+++ gdm-2.29.92.new/data/gconf.path	2010-03-23 09:33:35.164858194 +0100
@@ -6,3 +6,9 @@
 # owned by GDM.  Sysadmins should create
 # another source if they wish to override them.
 xml:readonly:$(HOME)/.gconf.mandatory
+
+# user values
+xml:readwrite:$(HOME)/.gconf
+
+# distribution default values
+xml:readwrite:$(HOME)/.gconf.defaults
