Index: gdm-2.32.0/common/gdm-common.c
===================================================================
--- gdm-2.32.0.orig/common/gdm-common.c	2010-09-15 20:58:10.000000000 +0200
+++ gdm-2.32.0/common/gdm-common.c	2010-12-07 15:21:02.907502001 +0100
@@ -47,6 +47,7 @@
 gboolean
 gdm_is_version_unstable (void)
 {
+#if 0
         char   **versions;
         gboolean unstable;
 
@@ -63,6 +64,9 @@
         g_strfreev (versions);
 
         return unstable;
+#else
+        return FALSE;
+#endif
 }
 
 void
