Index: gdm-3.0.0/data/Init.in
===================================================================
--- gdm-3.0.0.orig/data/Init.in	2011-05-30 17:35:50.855075638 +0200
+++ gdm-3.0.0/data/Init.in	2011-05-30 17:35:57.835075641 +0200
@@ -50,7 +50,7 @@
   UNAME=`gdmwhich uname`
   PROCESSOR=`$UNAME -p`
   if [ "x$PROCESSOR" = "xsparc" ]; then
-    if $XMODMAP | /usr/bin/grep mod4 | /usr/bin/grep Alt > /dev/null 2>/dev/null
+    if $XMODMAP | grep mod4 | grep Alt > /dev/null 2>/dev/null
     then
       $XMODMAP -e "clear Mod1" \
                -e "clear Mod4" \
