Description: Unify power related strings in the UI
Author: Ted Gould <ted.gould@canonical.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/531493
Forwarded: no

=== modified file 'gui/simple-greeter/gdm-greeter-panel.c'
--- old/gui/simple-greeter/gdm-greeter-panel.c	2009-12-04 16:19:35 +0000
+++ new/gui/simple-greeter/gdm-greeter-panel.c	2010-03-03 16:13:14 +0000
@@ -942,7 +942,7 @@
                         gtk_menu_shell_append (GTK_MENU_SHELL (panel->priv->shutdown_menu), menu_item);
                 } else {
                         if (can_suspend ()) {
-                                menu_item = gtk_menu_item_new_with_label (_("Suspend"));
+                                menu_item = gtk_menu_item_new_with_label (_("Sleep"));
                                 g_signal_connect (G_OBJECT (menu_item), "activate", G_CALLBACK (do_system_suspend), NULL);
                                 gtk_menu_shell_append (GTK_MENU_SHELL (panel->priv->shutdown_menu), menu_item);
                         }
