Description: Alters the way Xsession handles language settings.
Bug-Ubuntu: https://bugs.launchpad.net/bugs/553162
Forwarded: not-needed
Author: Gunnar Hjalmarsson <ubuntu@gunnar.cc>

diff -Nur -x '*.orig' -x '*~' gdm-2.32.0//data/Xsession.in gdm-2.32.0.new//data/Xsession.in
--- gdm-2.32.0//data/Xsession.in	2010-12-21 16:05:35.622679001 +0100
+++ gdm-2.32.0.new//data/Xsession.in	2010-12-24 06:23:01.552679001 +0100
@@ -146,11 +146,78 @@
 
 unset XKB_IN_USE
 
-# Overwrite $LANG from /etc/profile (and friends) with the one picked in the
-# greeter
-if [ -n "$GDM_LANG" ]; then
-  LANG="$GDM_LANG"
-  export LANG
+language_environment_update() {
+    if [ -z "$langlist" ] || [ $gdm_selected != ${langlist%%:*} ]; then
+        # make the first value in the LANGUAGE priority list correspond
+        # with the language selected in the GDM greeter
+        if [ -n "$langlist" -a ${gdm_selected%%[_@]*} != 'en' ]; then
+            if expr $langlist : ".*:$gdm_selected:" > /dev/null ; then
+                langlist=$( echo $langlist | sed -r "s/(.+:)($gdm_selected:)/\2\1/" )
+            else
+                langlist=$gdm_selected:$langlist
+            fi
+        else
+            # build list with 'en' as the last element
+            # no other languages if English was selected
+            langlist=$gdm_selected
+            langcode=${gdm_selected%%[_@]*}
+            [ $langlist != $langcode ] && langlist=$langlist:$langcode
+            [ $langcode != 'en' ] && langlist=$langlist:en_GB:en
+        fi
+    fi
+
+    # set LC_MESSAGES for applications that don't recognize LANGUAGE
+    lcmess=
+    locales=$( locale -a | grep -F .utf8 )
+    for loc in $locales; do
+        if [ $gdm_selected = "${loc%.*}${loc#*.utf8}" ]; then
+            lcmess=$loc
+            break
+        fi
+    done
+    if [ -z "$lcmess" ]; then
+        for loc in $locales; do
+            if [ ${gdm_selected%%[_@]*} = ${loc%%[._@]*} ]; then
+                lcmess=$loc
+                break
+            fi
+        done
+    fi
+    test -n "$lcmess" || lcmess='en_GB.utf8'
+
+    dmrc_update() {
+        key=$1; value=$2
+        if [ "$( grep -F $key $dmrc )" ]; then
+            sed -i "s/$key.*/$key=$value/" $dmrc
+        else
+            echo "$key=$value" >> $dmrc
+        fi
+    }
+    dmrc_update 'Langlist' $langlist
+    dmrc_update 'LCMess' $lcmess
+}
+
+# read /var/cache/gdm/$USER/dmrc and set LANGUAGE and LC_MESSAGES
+if [ -n "$GDM_LANG" -a $GDMSESSION != 'guest-restricted' ]; then
+    if [ $GDM_LANG != ${GDM_LANG%.utf8*} ]; then
+        gdm_selected=${GDM_LANG%.*}${GDM_LANG#*.utf8}
+    else
+        gdm_selected=$GDM_LANG
+    fi
+    dmrc=/var/cache/gdm/$USER/dmrc
+    test -f "$dmrc" || {
+        echo "\n[Desktop]\nLanguage=$GDM_LANG" > $dmrc
+    }
+    while read line; do
+        [ "${line%=*}" = 'Langlist' ] && langlist=${line#*=}
+        [ "${line%=*}" = 'LCMess' ] && lcmess=${line#*=}
+    done < $dmrc
+    if [ -z "$langlist" -a -z "$lcmess" ]; then
+        # Both fields empty => we assume that this is $USER's first login since upgrading
+        # from a pre 2.32.0-0ubuntu2 GDM version, so we fake $gdm_selected in an attempt to
+        # get the language environment right from start and prevent an undesired surprise.
+        if [ -n "$LANGUAGE" ]; then
+            gdm_selected=${LANGUAGE%%:*}
+            greeter_value=$( echo $gdm_selected | sed -r 's/([^@]+)(.*)/\1.utf8\2/' )
+            if [ "$( grep -F Language $dmrc )" ]; then
+                sed -i "s/Language=.*/Language=$greeter_value/" $dmrc
+            else
+                echo "Language=$greeter_value" >> $dmrc
+            fi
+        fi
+    fi
+    if [ -z "$langlist" ] || [ $gdm_selected != ${langlist%%:*} ]; then
+        # If the user changed language, update the env. values and edit the cache file.
+        language_environment_update
+    fi
+    export LANGUAGE="$langlist"
+    export LC_MESSAGES="$lcmess"
 fi
 
 # run all system xinitrc shell scripts.
