From eef9fec895f1411066cb15c64be10e5e84b565e0 Mon Sep 17 00:00:00 2001
From: Brian Cameron <brian.cameron@oracle.com>
Date: Thu, 02 Jun 2011 14:56:41 +0000
Subject: Only set WINDOWPATH if it is not NULL.

---
diff --git a/daemon/gdm-session-direct.c b/daemon/gdm-session-direct.c
index d1fec90..73c34e2 100644
--- a/daemon/gdm-session-direct.c
+++ b/daemon/gdm-session-direct.c
@@ -2047,6 +2047,8 @@ gdm_session_direct_set_environment_variable (GdmSessionDirect *session,
 static void
 setup_session_environment (GdmSessionDirect *session)
 {
+        char *windowpath;
+
         gdm_session_direct_set_environment_variable (session,
                                                      "GDMSESSION",
                                                      get_session_name (session));
@@ -2078,10 +2080,12 @@ setup_session_environment (GdmSessionDirect *session)
                                                              session->priv->user_x11_authority_file);
         }
 
-        gdm_session_direct_set_environment_variable (session,
-                                                     "WINDOWPATH",
-                                                     g_getenv ("WINDOWPATH"));
-
+        windowpath = g_getenv ("WINDOWPATH");
+        if (windowpath != NULL) {
+                gdm_session_direct_set_environment_variable (session,
+                                                             "WINDOWPATH",
+                                                             windowpath);
+        }
 
         /* FIXME: We do this here and in the session worker.  We should consolidate
          * somehow.
--
cgit v0.9.0.2
