Index: gdm-2.32.1/gui/simple-greeter/gdm-greeter-panel.c
===================================================================
--- gdm-2.32.1.orig/gui/simple-greeter/gdm-greeter-panel.c	2011-04-08 16:34:14.431921366 +1000
+++ gdm-2.32.1/gui/simple-greeter/gdm-greeter-panel.c	2011-04-08 16:34:18.581921368 +1000
@@ -889,7 +889,7 @@
 gdm_greeter_panel_init (GdmGreeterPanel *panel)
 {
         panel->priv = GDM_GREETER_PANEL_GET_PRIVATE (panel);
-
+        gtk_window_set_has_resize_grip (GTK_WINDOW (panel), FALSE);
 }
 
 static void
