Index: gdm-2.32.0/daemon/gdm-user-manager.c
===================================================================
--- gdm-2.32.0.orig/daemon/gdm-user-manager.c	2011-01-27 23:13:58.957847000 +0100
+++ gdm-2.32.0/daemon/gdm-user-manager.c	2011-01-27 23:14:07.907847000 +0100
@@ -72,7 +72,7 @@
 #ifdef __sun
 #define DEFAULT_MINIMAL_UID     100
 #else
-#define DEFAULT_MINIMAL_UID     500
+#define DEFAULT_MINIMAL_UID     1000
 #endif
 
 #ifndef _PATH_SHELLS
