--- gdm-2.16.4.orig/gui/gdmchooser.c
+++ gdm-2.16.4/gui/gdmchooser.c
@@ -448,6 +448,14 @@
     int pipe_buf;
     gboolean host_not_willing = FALSE;
 
+#ifdef PIPE_BUF
+    pipe_buf = PIPE_BUF;
+#else
+    /* apparently Hurd doesn't have PIPE_BUF */
+    pipe_buf = fpathconf (1 /*stdout*/, _PC_PIPE_BUF);
+    /* could return -1 if no limit */
+#endif
+
     if ( ! (condition & G_IO_IN)) 
         return TRUE;
 
@@ -508,7 +516,7 @@
 
 	    hostname = hbuf;
 
-	    if (strlen (hostname)+1 > PIPE_BUF)
+	    if (strlen (hostname) + 1 > pipe_buf)
 		goto done;
 	    hostname = g_strdup (hostname);
 
@@ -531,7 +539,7 @@
 				AF_INET);
 
 	    hostname = (he && he->h_name) ? he->h_name : inet_ntoa (clnt_sa.sin_addr);
-	    if (strlen (hostname)+1 > PIPE_BUF)
+	    if (strlen (hostname) + 1 > pipe_buf)
                    goto done;
 
 	    hostname = g_strdup (hostname);
@@ -544,14 +552,6 @@
         }
     }
 
-#ifdef PIPE_BUF
-    pipe_buf = PIPE_BUF;
-#else
-    /* apparently Hurd doesn't have PIPE_BUF */
-    pipe_buf = fpathconf (1 /*stdout*/, _PC_PIPE_BUF);
-    /* could return -1 if no limit */
-#endif
-
     /* We can't pipe hostnames larger than this */
     if (pipe_buf > 0 && strlen (hostname)+1 > pipe_buf) {
 	    g_free (hostname);
