Index: gdm-2.20.1/daemon/gdm.c
===================================================================
--- gdm-2.20.1.orig/daemon/gdm.c	2007-11-18 16:39:51.000000000 +0100
+++ gdm-2.20.1/daemon/gdm.c	2007-11-18 16:39:59.000000000 +0100
@@ -1585,9 +1585,7 @@
 
 	main_loop = g_main_loop_new (NULL, FALSE);
 
-	if (setlocale (LC_CTYPE, NULL) != NULL) {
-		gdm_system_locale = g_strdup (setlocale (LC_CTYPE, NULL));
-	}
+	gdm_system_locale = g_strdup (g_get_language_names()[0]);
 	/* initial TERM/INT handler */
 	sig.sa_handler = initial_term_int;
 	sig.sa_flags = SA_RESTART;
Index: gdm-2.20.1/daemon/slave.c
===================================================================
--- gdm-2.20.1.orig/daemon/slave.c	2007-11-18 16:39:51.000000000 +0100
+++ gdm-2.20.1/daemon/slave.c	2007-11-18 16:39:59.000000000 +0100
@@ -5576,7 +5576,7 @@
 				const char *gdm_system_locale;
 
 				locale = (gchar*)(msg + 3);
-				gdm_system_locale = setlocale (LC_CTYPE, NULL);
+				gdm_system_locale = g_get_language_names()[0];
 
 				always_restart_greeter = (gboolean)(*(msg + 2));
 				ve_clearenv ();
