Index: gdm-2.16.4/config/gdm.conf.in
===================================================================
--- gdm-2.16.4.orig/config/gdm.conf.in	2007-05-01 18:21:48.000000000 +0200
+++ gdm-2.16.4/config/gdm.conf.in	2007-05-01 18:23:25.000000000 +0200
@@ -93,9 +93,9 @@
 # always true and this configuration setting is ignored.
 #AlwaysRestartServer=false
 
-# User and group used for running GDM GUI applicaitons.  By default this is set
+# User and group used for running GDM GUI applications.  By default this is set
 # to user "gdm" and group "gdm".  This user/group should have very limited
-# permissions and access to ony the gdm directories and files.
+# permissions and access to only the gdm directories and files.
 User=gdm
 Group=gdm
 
@@ -160,7 +160,7 @@
 # The X nest command.
 Xnest=@X_XNEST_PATH@/Xnest @X_XNEST_CONFIG_OPTIONS@
 # Automatic VT allocation.  Right now only works on Linux.  This way we force
-# X to use specific vts.  turn VTAllocation to false if this is causing
+# X to use specific vts.  Turn VTAllocation to false if this is causing
 # problems.
 FirstVT=7
 VTAllocation=true
@@ -247,7 +247,7 @@
 
 # XDMCP is the protocol that allows remote login.  If you want to log into GDM
 # remotely (I'd never turn this on on open network, use ssh for such remote
-# usage that).  You can then run X with -query <thishost> to log in, or
+# usage).  You can then run X with -query <thishost> to log in, or
 # -indirect <thishost> to run a chooser.  Look for the 'Terminal' server type
 # at the bottom of this config file.
 [xdmcp]
