GNOME #436814

Index: gdm-2.18.1/config/gdm.conf.in
===================================================================
--- gdm-2.18.1.orig/config/gdm.conf.in	2007-05-02 10:38:10.000000000 +0200
+++ gdm-2.18.1/config/gdm.conf.in	2007-05-02 11:08:29.000000000 +0200
@@ -88,9 +88,9 @@
 # always true and this configuration setting is ignored.
 #AlwaysRestartServer=@ALWAYS_RESTART_SERVER@
 
-# User and group used for running GDM GUI applicaitons.  By default this is set
+# User and group used for running GDM GUI applications.  By default this is set
 # to user "gdm" and group "gdm".  This user/group should have very limited
-# permissions and access to ony the gdm directories and files.
+# permissions and access to only the gdm directories and files.
 User=gdm
 Group=gdm
 
@@ -166,7 +166,7 @@
 XnestUnscaledFontPath=@X_XNEST_UNSCALED_FONTPATH@
 
 # Automatic VT allocation.  Right now only works on Linux.  This way we force
-# X to use specific vts.  turn VTAllocation to false if this is causing
+# X to use specific vts.  Turn VTAllocation to false if this is causing
 # problems.
 FirstVT=7
 VTAllocation=true
@@ -255,7 +255,7 @@
 
 # XDMCP is the protocol that allows remote login.  If you want to log into GDM
 # remotely (I'd never turn this on on open network, use ssh for such remote
-# usage that).  You can then run X with -query <thishost> to log in, or
+# usage).  You can then run X with -query <thishost> to log in, or
 # -indirect <thishost> to run a chooser.  Look for the 'Terminal' server type
 # at the bottom of this config file.
 [xdmcp]
