Index: gdm-3.2.1.1/daemon/gdm-welcome-session.c
===================================================================
--- gdm-3.2.1.1.orig/daemon/gdm-welcome-session.c	2011-10-19 17:41:52.000000000 +0200
+++ gdm-3.2.1.1/daemon/gdm-welcome-session.c	2012-02-20 23:05:20.151709517 +0100
@@ -323,6 +323,7 @@ build_welcome_environment (GdmWelcomeSes
         g_hash_table_insert (hash, g_strdup ("RUNNING_UNDER_GDM"), g_strdup ("true"));
         g_hash_table_insert (hash, g_strdup ("GVFS_DISABLE_FUSE"), g_strdup ("1"));
         g_hash_table_insert (hash, g_strdup ("DCONF_PROFILE"), g_strdup ("gdm"));
+        g_hash_table_insert (hash, g_strdup ("DCONF_DIR"), g_strdup ("/var/lib/gdm3/dconf"));
 
         return hash;
 }
