--- a/daemon/gdm-launch-environment.c
+++ b/daemon/gdm-launch-environment.c
@@ -289,6 +289,7 @@
 
         g_hash_table_insert (hash, g_strdup ("RUNNING_UNDER_GDM"), g_strdup ("true"));
         g_hash_table_insert (hash, g_strdup ("DCONF_PROFILE"), g_strdup ("gdm"));
+        g_hash_table_insert (hash, g_strdup ("DCONF_DIR"), g_strdup ("/var/lib/gdm3/dconf"));
 
         if (launch_environment->priv->dbus_session_bus_address) {
                 g_hash_table_insert (hash, g_strdup ("DBUS_SESSION_BUS_ADDRESS"), g_strdup (launch_environment->priv->dbus_session_bus_address));
