Index: b/data/gdm.service.in
===================================================================
--- a/data/gdm.service.in
+++ b/data/gdm.service.in
@@ -20,15 +20,16 @@
 OnFailure=plymouth-quit.service
 
 [Service]
-ExecStart=@sbindir@/gdm
+ExecStartPre=/bin/sh -c '[ "$(cat /etc/X11/default-display-manager 2>/dev/null)" = "/usr/sbin/gdm3" ]'
+ExecStartPre=/usr/share/gdm/generate-config
+ExecStart=@sbindir@/gdm3
 KillMode=mixed
 Restart=always
+RestartSec=1s
 IgnoreSIGPIPE=no
 BusName=org.gnome.DisplayManager
 StandardOutput=syslog
 StandardError=inherit
 EnvironmentFile=-@LANG_CONFIG_FILE@
 ExecReload=/bin/kill -SIGHUP $MAINPID
-
-[Install]
-Alias=display-manager.service
+ExecReload=/usr/share/gdm/generate-config
