--- a/data/gdm.service.in
+++ b/data/gdm.service.in
@@ -20,15 +20,15 @@ After=rc-local.service plymouth-start.se
 OnFailure=plymouth-quit.service
 
 [Service]
-ExecStart=@sbindir@/gdm
+ExecStartPre=/usr/share/gdm/generate-config
+ExecStart=@sbindir@/gdm3
 KillMode=mixed
 Restart=always
+RestartSec=1s
 IgnoreSIGPIPE=no
 BusName=org.gnome.DisplayManager
 StandardOutput=syslog
 StandardError=inherit
 EnvironmentFile=-@LANG_CONFIG_FILE@
+ExecReload=/usr/share/gdm/generate-config
 ExecReload=/bin/kill -SIGHUP $MAINPID
-
-[Install]
-Alias=display-manager.service
