--- a/gui/libgdm/Makefile.am
+++ b/gui/libgdm/Makefile.am
@@ -41,7 +41,7 @@
         -DDATADIR=\""$(datadir)"\"                                            \
         $(END_OF_LIST)
 
-lib_LTLIBRARIES =                                                             \
+pkglib_LTLIBRARIES =                                                   \
         libgdm.la                                                      \
         $(END_OF_LIST)
 
@@ -88,9 +88,9 @@
 -include $(INTROSPECTION_MAKEFILE)
 
 if HAVE_INTROSPECTION
-girdir = $(datadir)/gir-1.0
+girdir = $(pkglibdir)
 gir_DATA = Gdm-1.0.gir
-typelibsdir = $(libdir)/girepository-1.0
+typelibsdir = $(pkglibdir)
 typelibs_DATA = $(gir_DATA:.gir=.typelib)
 INTROSPECTION_GIRS = $(gir_DATA)
 
