Index: gdm-3.5.4.2~git20120803~00576a2/gui/simple-greeter/libgdmsimplegreeter/Makefile.am
===================================================================
--- gdm-3.5.4.2~git20120803~00576a2.orig/gui/simple-greeter/libgdmsimplegreeter/Makefile.am	2012-08-05 16:30:04.000000000 +0200
+++ gdm-3.5.4.2~git20120803~00576a2/gui/simple-greeter/libgdmsimplegreeter/Makefile.am	2012-08-05 17:44:36.000000000 +0200
@@ -14,7 +14,7 @@
 	$(GTK_CFLAGS)				\
 	$(NULL)
 
-lib_LTLIBRARIES = 			\
+pkglib_LTLIBRARIES = 			\
 	libgdmsimplegreeter.la		\
 	$(NULL)
 
Index: gdm-3.5.4.2~git20120803~00576a2/gui/libgdm/Makefile.am
===================================================================
--- gdm-3.5.4.2~git20120803~00576a2.orig/gui/libgdm/Makefile.am	2012-08-05 16:30:04.000000000 +0200
+++ gdm-3.5.4.2~git20120803~00576a2/gui/libgdm/Makefile.am	2012-08-05 17:50:16.000000000 +0200
@@ -40,7 +40,7 @@
         -DDATADIR=\""$(datadir)"\"                                            \
         $(END_OF_LIST)
 
-lib_LTLIBRARIES =                                                             \
+pkglib_LTLIBRARIES =                                                   \
         libgdm.la                                                      \
         $(END_OF_LIST)
 
@@ -83,9 +83,9 @@
 -include $(INTROSPECTION_MAKEFILE)
 
 if HAVE_INTROSPECTION
-girdir = $(datadir)/gir-1.0
+girdir = $(pkglibdir)
 gir_DATA = Gdm-1.0.gir
-typelibsdir = $(libdir)/girepository-1.0
+typelibsdir = $(pkglibdir)
 typelibs_DATA = $(gir_DATA:.gir=.typelib)
 INTROSPECTION_GIRS = $(gir_DATA)
 
