From: Jeremy Bicha <jbicha@ubuntu.com>
Date: Sat, 4 Aug 2018 22:08:27 +0200
Subject: Fix 'make check' in po/

Forwarded: not-needed
Origin: vendor, Debian
---
 po/POTFILES.skip | 1 +
 1 file changed, 1 insertion(+)

diff --git a/po/POTFILES.skip b/po/POTFILES.skip
index 8b7f390..84c4f48 100644
--- a/po/POTFILES.skip
+++ b/po/POTFILES.skip
@@ -7,3 +7,4 @@
 # remove these unless you fix "make distcheck" a different way.
 #
 data/gdm.schemas.in
+debian/default.desktop.in
