Description: Run /etc/X11/Xsession.d/.
Author: Martin Pitt <martin.pitt@ubuntu.com>
Origin: Ubuntu; /etc/X11/Xsession.d is a Debianism/Ubuntuism

Index: gdm-3.7.2/data/Xsession.in
===================================================================
--- gdm-3.7.2.orig/data/Xsession.in	2012-05-04 10:51:06.000000000 +1200
+++ gdm-3.7.2/data/Xsession.in	2012-12-04 10:38:43.183982014 +1300
@@ -164,6 +164,19 @@
     done
 fi
 
+# run /etc/X11/Xsession.d/
+OPTIONFILE=/etc/X11/Xsession.options
+USERXSESSION=$HOME/.xsession
+USERXSESSIONRC=$HOME/.xsessionrc
+ALTUSERXSESSION=$HOME/.Xsession
+if [ -d /etc/X11/Xsession.d ]; then
+    for i in `ls /etc/X11/Xsession.d/` ; do
+        if [ -r "/etc/X11/Xsession.d/$i"  -a -f "/etc/X11/Xsession.d/$i" ] && expr "$i" : '^[[:alnum:]_-]\+$' > /dev/null; then
+	    . "/etc/X11/Xsession.d/$i"
+        fi
+    done
+fi
+
 if [ "x$command" = "xdefault" ] ; then
   if [ -x "$HOME/.Xclients" ]; then
     command="$HOME/.Xclients"

